<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="zjanna" />

<meta name="date" content="2019-06-07" />

<title>pancreas</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">characterize_batches</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/almutlue/characterize_batches">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">pancreas</h1>
<h4 class="author">zjanna</h4>
<h4 class="date">2019-06-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-07-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>characterize_batches/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongobjectspresent"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Environment:</strong> objects present </a>
</p>
</div>
<div id="strongEnvironmentstrongobjectspresent" class="panel-collapse collapse">
<div class="panel-body">
<p>
The global environment had objects present when the code in the R Markdown file was run. These objects can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment. Use <code>wflow_publish</code> or <code>wflow_build</code> to ensure that the code is always run in an empty environment.
</p>
<p>
The following objects were defined in the global environment when these results were created:
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Class
</th>
<th style="text-align:left;">
Size
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
data
</td>
<td style="text-align:left;">
environment
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
env
</td>
<td style="text-align:left;">
environment
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190528code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190528)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190528code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190528)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomalmutluecharacterizebatchestree2e5c1abfd292a050a60c7e5de29a6bcabadcb083targetblank2e5c1aba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/almutlue/characterize_batches/tree/2e5c1abfd292a050a60c7e5de29a6bcabadcb083" target="_blank">2e5c1ab</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomalmutluecharacterizebatchestree2e5c1abfd292a050a60c7e5de29a6bcabadcb083targetblank2e5c1aba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/diagnostic_plots.R
    Untracked:  data/PBMC_media_merged_clusters_seurat.rds
    Untracked:  data/csf_media.rds
    Untracked:  data/pancreas.rds
    Untracked:  data/pbmc2_media.rds
    Untracked:  data/pbmc_media.rds
    Untracked:  data/sce_csf_Integrated.rds
    Untracked:  data/sce_hca_integrated.rds
    Untracked:  data/sce_khang_patient.rds
    Untracked:  data/sce_pbmc_Integrated.rds
    Untracked:  deg_hca.rds
    Untracked:  iscb_v2.pdf
    Untracked:  limma_hca.rds
    Untracked:  output/cms_4_mouse_red.rds
    Untracked:  output/cms_all_mouse_red.rds
    Untracked:  output/cms_cellBench.rds
    Untracked:  output/cms_csf_media_storage.rds
    Untracked:  output/cms_csf_patient.rds
    Untracked:  output/cms_khang_patient.rds
    Untracked:  output/cms_pancreas.rds
    Untracked:  output/cms_pbmc2_media_storage.rds
    Untracked:  output/cms_pbmc2_patient.rds
    Untracked:  output/cms_pbmc_media_storage.rds
    Untracked:  output/d_exp_cellBench.rds
    Untracked:  output/d_exp_csf_media_storage.rds
    Untracked:  output/d_exp_csf_patient.rds
    Untracked:  output/d_exp_ctcellBench.rds
    Untracked:  output/d_exp_ctcsf_media_storage.rds
    Untracked:  output/d_exp_ctcsf_patient.rds
    Untracked:  output/d_exp_cthca.rds
    Untracked:  output/d_exp_ctkhang_patient.rds
    Untracked:  output/d_exp_ctpancreas.rds
    Untracked:  output/d_exp_ctpbmc2_media_storage.rds
    Untracked:  output/d_exp_ctpbmc2_patient.rds
    Untracked:  output/d_exp_ctpbmc_media_storage.rds
    Untracked:  output/d_exp_hca.rds
    Untracked:  output/d_exp_khang_patient.rds
    Untracked:  output/d_exp_pancreas.rds
    Untracked:  output/d_exp_pbmc2_media_storage.rds
    Untracked:  output/d_exp_pbmc2_patient.rds
    Untracked:  output/d_exp_pbmc_media_storage.rds
    Untracked:  output/deg_cellBench.rds
    Untracked:  output/deg_csf_media_storage.rds
    Untracked:  output/deg_csf_patient.rds
    Untracked:  output/deg_hca.rds
    Untracked:  output/deg_khang_patient.rds
    Untracked:  output/deg_pancreas.rds
    Untracked:  output/deg_pbmc2_media_storage.rds
    Untracked:  output/deg_pbmc2_patient.rds
    Untracked:  output/deg_pbmc_media_storage.rds
    Untracked:  output/integrated_protocols_mouse_red.rds
    Untracked:  output/limma_cellBench.rds
    Untracked:  output/limma_csf_media_storage.rds
    Untracked:  output/limma_csf_patient.rds
    Untracked:  output/limma_hca.rds
    Untracked:  output/limma_khang_patient.rds
    Untracked:  output/limma_pancreas.rds
    Untracked:  output/limma_pbmc2_media_storage.rds
    Untracked:  output/limma_pbmc2_patient.rds
    Untracked:  output/limma_pbmc_media_storage.rds
    Untracked:  output/lrt_ct_csf_media_storage.rds
    Untracked:  output/lrt_ct_csf_patient.rds
    Untracked:  output/pbmc_media_cms.rds
    Untracked:  output/pbmc_media_limma_results.RData
    Untracked:  output/pbmc_media_upset_data.RData
    Untracked:  output/protocol_processed_sce_list_red.rds
    Untracked:  output/seurat_pbmc_Integrated.rds
    Untracked:  output/summary_cellBench.rds
    Untracked:  output/summary_csf_media_storage.rds
    Untracked:  output/summary_csf_patient.rds
    Untracked:  output/summary_hca.rds
    Untracked:  output/summary_khang_patient.rds
    Untracked:  output/summary_pancreas.rds
    Untracked:  output/summary_pbmc2_media_storage.rds
    Untracked:  output/summary_pbmc2_patient.rds
    Untracked:  output/summary_pbmc_media.rds
    Untracked:  output/summary_pbmc_media_storage.rds
    Untracked:  output/upset_cellBench.rds
    Untracked:  output/upset_csf_media_storage.rds
    Untracked:  output/upset_csf_patient.rds
    Untracked:  output/upset_hca.rds
    Untracked:  output/upset_khang_patient.rds
    Untracked:  output/upset_pancreas.rds
    Untracked:  output/upset_pbmc2_media_storage.rds
    Untracked:  output/upset_pbmc2_patient.rds
    Untracked:  output/upset_pbmc_media_storage.rds
    Untracked:  output/vp_cellBench.rds
    Untracked:  output/vp_csf_media_storage.rds
    Untracked:  output/vp_csf_patient.rds
    Untracked:  output/vp_hca.rds
    Untracked:  output/vp_khang_patient.rds
    Untracked:  output/vp_pancreas.rds
    Untracked:  output/vp_pbmc.rds
    Untracked:  output/vp_pbmc2_media_storage.rds
    Untracked:  output/vp_pbmc2_patient.rds
    Untracked:  output/vp_pbmc_media_storage.rds

Unstaged changes:
    Modified:   analysis/hca.Rmd
    Modified:   analysis/khan.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/characterize_batches/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/pancreas.html" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f7d5ec3a5298ad9fcd20c19129e4f840b27e9487/analysis/pancreas.Rmd" target="_blank">f7d5ec3</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-28
</td>
<td>
minor add for cellbench
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3c4c8fa577603f4e17b6f63c45bed3046acaa0ca/analysis/pancreas.Rmd" target="_blank">3c4c8fa</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-27
</td>
<td>
add ternary plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/8259344c430674125d5ecec54924e03a73c977f0/analysis/pancreas.Rmd" target="_blank">8259344</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-27
</td>
<td>
new_pancreas.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/00723e90561493e77c31044d5cf469f2de97b8ea/analysis/pancreas.Rmd" target="_blank">00723e9</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-27
</td>
<td>
uncomment pierre part
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/30b0e7c2543fc5e25ecc452777a065ac83ab643f/analysis/pancreas.Rmd" target="_blank">30b0e7c</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-27
</td>
<td>
paths
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/acb824b71493f17c3cc7e5801ede262ff9e2ff3d/analysis/pancreas.Rmd" target="_blank">acb824b</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-27
</td>
<td>
update files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/characterize_batches/acb824b71493f17c3cc7e5801ede262ff9e2ff3d/docs/pancreas.html" target="_blank">acb824b</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-27
</td>
<td>
update files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/characterize_batches/887b9be03a080d67f15bdcc2d170e9b94a4b2c9f/docs/pancreas.html" target="_blank">887b9be</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/characterize_batches/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/pancreas.html" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/characterize_batches/ddd4d10c222cdd77ac50f0ef61b860a589f912d7/docs/pancreas.html" target="_blank">ddd4d10</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/199c579b8e99dbbd7585b82d53902df43dd48875/analysis/pancreas.Rmd" target="_blank">199c579</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-19
</td>
<td>
wflow_publish(“analysis/pancreas.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/characterize_batches/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/pancreas.html" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/characterize_batches/b6183ee8ddc6f60fa8c698d18838b90142058f4a/docs/pancreas.html" target="_blank">b6183ee</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/1f75e4f4407558fb1285e5abce16a264d677f42f/analysis/pancreas.Rmd" target="_blank">1f75e4f</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-11
</td>
<td>
wflow_publish(“analysis/pancreas.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/characterize_batches/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/pancreas.html" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/ef1e0a82e8f6e79efc74d9027851cd2f0149da37/analysis/pancreas.Rmd" target="_blank">ef1e0a8</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
<td>
wflow_publish(“analysis/pancreas.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="technical-batch-effects" class="section level1">
<h1>Technical batch effects</h1>
<p><strong>Pancreas data</strong>, test batch effect from different protocols. For the purposes of this example, it was chosen human pancreatic islet cell datasets produced across four technologies, CelSeq (GSE81076) CelSeq2 (GSE85241), Fluidigm C1 (GSE86469), and SMART-Seq2 (E-MTAB-5061).</p>
<pre class="r"><code>suppressPackageStartupMessages({
  library(CellBench)
  library(scater)
  library(jcolors)
  library(CellMixS)
  library(gridExtra)
  library(purrr)
  library(jcolors)
  library(here)
  library(tidyr)
  library(dplyr)
  library(stringr)
  library(diffcyt)
  library(ComplexHeatmap)
  library(variancePartition)
  library(ggtern)
  library(TruncatedDistributions)
  library(scran)
})</code></pre>
<p>Dataset</p>
<pre class="r"><code># data
data_path &lt;- here::here(&quot;data&quot;)
out_path &lt;- here::here(&quot;output&quot;)
code_path &lt;- here::here(&quot;code&quot;)
dataset_name &lt;- &quot;pancreas&quot;

sce &lt;- readRDS(paste0(data_path,&quot;/&quot;,  &quot;pancreas.rds&quot;))
sce &lt;- computeSumFactors(sce)
sce &lt;- normalize(sce)
sce &lt;- runUMAP(sce)

# colors
cols &lt;-c(c(jcolors(&#39;pal6&#39;),jcolors(&#39;pal8&#39;))[c(1,8,14,5,2:4,6,7,9:13,15:20)],jcolors(&#39;pal4&#39;))
names(cols) &lt;- c()

# param
MultiSample = FALSE

#variables
batch &lt;- &quot;tech&quot;
celltype &lt;- &quot;celltype&quot;
sample &lt;- NA

table(colData(sce)[, batch])</code></pre>
<pre><code>
   celseq   celseq2 smartseq2 
      976      2181      2202 </code></pre>
<pre class="r"><code>#contrast
cont&lt;-list(&quot;celseq-celseq2&quot; = c(1,-1,0),
           &quot;celseq-smartseq2&quot; = c(1,0,-1),
           &quot;celseq2-smartseq2&quot; = c(0,1,-1))

n_genes &lt;- nrow(sce)
n_cells &lt;- ncol(sce)</code></pre>
</div>
<div id="visualize-batch-effect" class="section level1">
<h1>Visualize batch effect</h1>
<pre class="r"><code>feature_list &lt;- c(batch, celltype)

lapply(feature_list, function(feature_name){
  visGroup(sce, feature_name, dim_red= &quot;UMAP&quot;)
})</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/pancreas.Rmd/vis%20batch-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-batch-1">
Past versions of “vis batch-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-batch-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/vis batch-1.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/vis batch-1.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/ddd4d10c222cdd77ac50f0ef61b860a589f912d7/docs/figure/pancreas.Rmd/vis batch-1.png" target="_blank">ddd4d10</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/vis batch-1.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/b6183ee8ddc6f60fa8c698d18838b90142058f4a/docs/figure/pancreas.Rmd/vis batch-1.png" target="_blank">b6183ee</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/vis batch-1.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/pancreas.Rmd/vis%20batch-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-batch-2">
Past versions of “vis batch-2.png”
</button>
</p>
<div id="fig-no-spaces-vis-batch-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/vis batch-2.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/vis batch-2.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/ddd4d10c222cdd77ac50f0ef61b860a589f912d7/docs/figure/pancreas.Rmd/vis batch-2.png" target="_blank">ddd4d10</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/vis batch-2.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/b6183ee8ddc6f60fa8c698d18838b90142058f4a/docs/figure/pancreas.Rmd/vis batch-2.png" target="_blank">b6183ee</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/vis batch-2.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sizestrength-of-the-batch-effcet" class="section level1">
<h1>Size/Strength of the batch effcet</h1>
<p>How much of the variance within data can be attributed to the batch effect?</p>
<div id="variancepartitioning" class="section level2">
<h2>VariancePartitioning</h2>
<pre class="r"><code>expr &lt;- as.matrix(assays(sce)$logcounts)
meta_sub &lt;- as.data.frame(colData(sce)[, c(celltype, batch)])
form &lt;- as.formula(paste0(&quot;~ (1|&quot;, celltype, &quot;) + (1|&quot;, batch, &quot;)&quot;))

#varPart &lt;- fitExtractVarPartModel(expr, form, meta_sub)

#Sort variables (i.e. columns) by median fraction# of variance explained
#vp &lt;- varPart
#saveRDS(vp, paste0(out_path,&quot;/vp_&quot;, dataset_name, &quot;.rds&quot;))
vp &lt;- readRDS(paste0(out_path,&quot;/vp_&quot;, dataset_name, &quot;.rds&quot;))

vp_names &lt;- rownames(vp)
vp &lt;-vp  %&gt;% dplyr::mutate(gene= vp_names) %&gt;% dplyr::arrange(-!! rlang::parse_expr(batch))</code></pre>
<pre><code>Warning in class(x) &lt;- c(subclass, tibble_class): Setze class(x) auf
mehrere Zeichenketten(&quot;tbl_df&quot;, &quot;tbl&quot;, ...); das Ergebnis ist kein S4
Objekt mehr</code></pre>
<pre class="r"><code>vp_sub &lt;- vp[1:3] %&gt;% set_rownames(vp$gene)

#plot
plotPercentBars( vp_sub[1:10,] )</code></pre>
<p><img src="figure/pancreas.Rmd/variance%20part-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-variance-part-1">
Past versions of “variance part-1.png”
</button>
</p>
<div id="fig-no-spaces-variance-part-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/variance part-1.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/variance part-1.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/b6183ee8ddc6f60fa8c698d18838b90142058f4a/docs/figure/pancreas.Rmd/variance part-1.png" target="_blank">b6183ee</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/variance part-1.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotVarPart( vp_sub )</code></pre>
<p><img src="figure/pancreas.Rmd/variance%20part-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-variance-part-2">
Past versions of “variance part-2.png”
</button>
</p>
<div id="fig-no-spaces-variance-part-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/variance part-2.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/variance part-2.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/variance part-2.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/b6183ee8ddc6f60fa8c698d18838b90142058f4a/docs/figure/pancreas.Rmd/variance part-2.png" target="_blank">b6183ee</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/variance part-2.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="deviance-explained" class="section level1">
<h1>Deviance explained</h1>
<div id="summarize-variance-partitioning" class="section level3">
<h3>Summarize variance partitioning</h3>
<pre class="r"><code>d_exp &lt;- readRDS(paste0(out_path,&quot;/d_exp_&quot;, dataset_name, &quot;.rds&quot;))
d_exp_ct &lt;- readRDS(paste0(out_path,&quot;/d_exp_ct&quot;, dataset_name, &quot;.rds&quot;))

## Summarize vp
# How many genes have a variance component affected by batch with &gt; 1%
n_batch_gene &lt;- as_tibble(vp) %&gt;%  
  dplyr::filter(!! rlang::parse_expr(batch) &gt; 0.01) %&gt;% nrow()/n_genes

n_batch_gene10 &lt;- as_tibble(vp) %&gt;%  
  dplyr::filter(!! rlang::parse_expr(batch) &gt; 0.1) %&gt;% nrow()/n_genes

n_celltype_gene &lt;- as_tibble(vp) %&gt;%  
  dplyr::filter(!! rlang::parse_expr(celltype)&gt; 0.01) %&gt;% nrow()/n_genes

n_rel &lt;- n_batch_gene/n_celltype_gene

## Sumarize deviance modeled
n_batch_gene_mod &lt;- length(which(d_exp &gt; 0.01))/n_genes
n_batch_gene10_mod &lt;- length(which(d_exp &gt; 0.1))/n_genes
n_celltype_gene_mod &lt;- length(which(d_exp_ct &gt; 0.01))/n_genes
n_rel_mod &lt;- n_batch_gene_mod/n_celltype_gene_mod

# Mean variance that is explained by the batch effect/celltype
m_batch &lt;- mean(asin(vp[, batch]))
m_celltype &lt;- mean(asin(vp[, celltype]))
m_rel &lt;- m_batch/m_celltype

m_batch_mod &lt;- mean(asin(d_exp))
m_celltype_mod &lt;- mean(asin(d_exp_ct))
m_rel_mod &lt;- m_batch_mod/m_celltype_mod


#Median variance explained by the batch effect/celltype
me_batch &lt;- median(asin(vp[, batch]))
me_celltype &lt;- median(asin(vp[, celltype]))
me_rel &lt;- me_batch/me_celltype

me_batch_mod &lt;- median(asin(d_exp))
me_celltype_mod &lt;- median(asin(d_exp_ct))
me_rel_mod &lt;- me_batch_mod/me_celltype_mod</code></pre>
</div>
</div>
<div id="plot-variancedeviance-and-gene-expression" class="section level1">
<h1>Plot variance/deviance and gene expression</h1>
<pre class="r"><code>#plot varinace dev-along with expression
rownames(vp) &lt;- vp$gene
vp &lt;- vp[rownames(sce),]
rowData(sce)$vp_batch &lt;- vp[,batch]
rowData(sce)$vp_ct &lt;- vp[,celltype]
rowData(sce)$vp_other &lt;- vp[,&quot;Residuals&quot;]

rowData(sce)$d_exp &lt;- d_exp
rowData(sce)$d_exp_ct &lt;- d_exp_ct
rowData(sce)$d_exp_other &lt;- 1 - d_exp - d_exp_ct
rowData(sce)$mean_expr &lt;- rowMeans(assays(sce)$logcounts)

#add expr classes 
th &lt;- quantile(rowMeans(assays(sce)$logcounts), c(.33, .66))
high_th &lt;- th[2]
mid_th &lt;- th[1]

rowData(sce)$expr_class &lt;- ifelse(rowMeans(assays(sce)$logcounts) &gt; high_th, &quot;high&quot;, ifelse(rowMeans(assays(sce)$logcounts) &lt;= high_th &amp; rowMeans(assays(sce)$logcounts) &gt; mid_th, &quot;medium&quot;, &quot;low&quot;))


plot_dev &lt;- function(var, var_col){
  ggplot(as.data.frame(rowData(sce)), aes_string(x = &quot;mean_expr&quot;, y = var, colour = var_col)) + 
  geom_point() + 
  geom_smooth(method = &quot;lm&quot;, se = FALSE)
}

plot_dev(&quot;vp_batch&quot;, &quot;d_exp&quot;)</code></pre>
<p><img src="figure/pancreas.Rmd/var%20dev%20expr-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-var-dev-expr-1">
Past versions of “var dev expr-1.png”
</button>
</p>
<div id="fig-no-spaces-var-dev-expr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/var dev expr-1.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_dev(&quot;d_exp&quot;, &quot;vp_batch&quot;)</code></pre>
<p><img src="figure/pancreas.Rmd/var%20dev%20expr-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-var-dev-expr-2">
Past versions of “var dev expr-2.png”
</button>
</p>
<div id="fig-no-spaces-var-dev-expr-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/var dev expr-2.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_dev(&quot;vp_ct&quot;, &quot;d_exp_ct&quot;)</code></pre>
<p><img src="figure/pancreas.Rmd/var%20dev%20expr-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-var-dev-expr-3">
Past versions of “var dev expr-3.png”
</button>
</p>
<div id="fig-no-spaces-var-dev-expr-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/var dev expr-3.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_dev(&quot;d_exp_ct&quot;, &quot;vp_ct&quot;)</code></pre>
<p><img src="figure/pancreas.Rmd/var%20dev%20expr-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-var-dev-expr-4">
Past versions of “var dev expr-4.png”
</button>
</p>
<div id="fig-no-spaces-var-dev-expr-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/var dev expr-4.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ternary-plots" class="section level1">
<h1>Ternary plots</h1>
<pre class="r"><code>ggtern(data=as.data.frame(rowData(sce)),aes(vp_batch, vp_ct, vp_other)) + 
  stat_density_tern(aes(fill=..level.., alpha=..level..),geom=&#39;polygon&#39;) +
  scale_fill_gradient2(high = &quot;red&quot;) +
  guides(color = &quot;none&quot;, fill = &quot;none&quot;, alpha = &quot;none&quot;) +
  geom_point(size= 0.1, alpha = 0.5)  + 
  theme_bw()</code></pre>
<pre><code>Warning: Removed 852 rows containing non-finite values (StatDensityTern).</code></pre>
<p><img src="figure/pancreas.Rmd/tern%20plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-tern-plots-1">
Past versions of “tern plots-1.png”
</button>
</p>
<div id="fig-no-spaces-tern-plots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/tern plots-1.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t1 &lt;- ggtern(data=as.data.frame(rowData(sce)),aes(vp_batch, vp_ct, vp_other)) + 
  geom_point(size = 0.1) +
  geom_density_tern() + 
  theme_bw()

t1</code></pre>
<pre><code>Warning: Removed 852 rows containing non-finite values (StatDensityTern).</code></pre>
<p><img src="figure/pancreas.Rmd/tern%20plots-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-tern-plots-2">
Past versions of “tern plots-2.png”
</button>
</p>
<div id="fig-no-spaces-tern-plots-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/tern plots-2.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t1 + facet_grid(~expr_class)</code></pre>
<pre><code>Warning: Removed 18 rows containing non-finite values (StatDensityTern).</code></pre>
<pre><code>Warning: Removed 758 rows containing non-finite values (StatDensityTern).</code></pre>
<pre><code>Warning: Removed 76 rows containing non-finite values (StatDensityTern).</code></pre>
<p><img src="figure/pancreas.Rmd/tern%20plots-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-tern-plots-3">
Past versions of “tern plots-3.png”
</button>
</p>
<div id="fig-no-spaces-tern-plots-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/tern plots-3.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggtern(data=as.data.frame(rowData(sce)),aes(d_exp, d_exp_ct, d_exp_other)) + 
  stat_density_tern(aes(fill=..level.., alpha=..level..),geom=&#39;polygon&#39;) +
  scale_fill_gradient2(high = &quot;red&quot;) +
  guides(color = &quot;none&quot;, fill = &quot;none&quot;, alpha = &quot;none&quot;) +
  geom_point(size= 0.1, alpha = 0.5) + 
  theme_bw() </code></pre>
<pre><code>Warning: Removed 11 rows containing non-finite values (StatDensityTern).</code></pre>
<p><img src="figure/pancreas.Rmd/tern%20plots-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-tern-plots-4">
Past versions of “tern plots-4.png”
</button>
</p>
<div id="fig-no-spaces-tern-plots-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/tern plots-4.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t2 &lt;- ggtern(data=as.data.frame(rowData(sce)),aes(d_exp, d_exp_ct, d_exp_other)) + 
  geom_point(size = 0.1) +
  geom_density_tern() + 
  theme_bw() 

t2</code></pre>
<pre><code>Warning: Removed 11 rows containing non-finite values (StatDensityTern).</code></pre>
<p><img src="figure/pancreas.Rmd/tern%20plots-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-tern-plots-5">
Past versions of “tern plots-5.png”
</button>
</p>
<div id="fig-no-spaces-tern-plots-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/tern plots-5.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t2 +  facet_grid(~expr_class)</code></pre>
<pre><code>Warning: Removed 8 rows containing non-finite values (StatDensityTern).</code></pre>
<pre><code>Warning: Removed 3 rows containing non-finite values (StatDensityTern).</code></pre>
<p><img src="figure/pancreas.Rmd/tern%20plots-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-tern-plots-6">
Past versions of “tern plots-6.png”
</button>
</p>
<div id="fig-no-spaces-tern-plots-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/tern plots-6.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="celltype-specificity" class="section level1">
<h1>Celltype specificity</h1>
<div id="celltype-abundance" class="section level2">
<h2>Celltype abundance</h2>
<pre class="r"><code>meta_tib &lt;- as_tibble(colData(sce)) %&gt;% group_by_at(c(batch, celltype)) %&gt;% summarize(n = n()) %&gt;% dplyr::mutate(cell_freq = n / sum(n))


plot_abundance &lt;- function(cluster_var, tib, x_var){
  meta_df &lt;- as.data.frame(eval(tib))
  p &lt;- ggplot(data=meta_df, aes_string(x=x_var, y=&quot;cell_freq&quot;, fill = cluster_var)) +
    geom_bar(stat=&quot;identity&quot;) + scale_fill_manual(values=cols)
  p + coord_flip() + theme_minimal()
}

plot_abundance(cluster_var = celltype, tib = meta_tib, x_var = batch)</code></pre>
<p><img src="figure/pancreas.Rmd/celltype%20abundance-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-celltype-abundance-1">
Past versions of “celltype abundance-1.png”
</button>
</p>
<div id="fig-no-spaces-celltype-abundance-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/celltype abundance-1.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/celltype abundance-1.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/celltype abundance-1.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/celltype abundance-1.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="celltype-abundance---relative-difference-in-abundances" class="section level2">
<h2>Celltype abundance - relative difference in abundances</h2>
<pre class="r"><code>meta_tib &lt;- as_tibble(colData(sce)) %&gt;% group_by_at(c(batch, celltype)) %&gt;% 
  summarize(n = n()) %&gt;% spread_(batch,&quot;n&quot;)
meta_df &lt;- as.data.frame(eval(meta_tib))[,-1]
meta_comb &lt;- combn(meta_df,2,simplify=FALSE)
res &lt;- lapply(meta_comb,function(x){
  cond1 &lt;- names(x)[1]
  cond2 &lt;- names(x)[2]
  rel_abund_diff &lt;- mapply(function(cond1, cond2)
    abs(cond1 - cond2)/(cond1 + cond2), x[, cond1], x[, cond2])
  rel_abund_diff
})

mean_rel_abund_diff &lt;- mean(unlist(res))
min_rel_abund_diff &lt;- min(unlist(res))
max_rel_abund_diff &lt;- max(unlist(res))</code></pre>
</div>
</div>
<div id="count-distribution" class="section level1">
<h1>Count distribution</h1>
<p>Do the overall counts distribution vary between batches? Clusterwise ploting?</p>
<pre class="r"><code>#sample an cluster ids
sids &lt;- levels(as.factor(colData(sce)[, batch]))
names(sids) &lt;- sids
cids &lt;- levels(as.factor(colData(sce)[, celltype]))
names(cids) &lt;- cids

#mean gene expression by sample and cluster
mean_list &lt;- lapply(sids, function(batch_var){
  mean_cluster &lt;- lapply(cids, function(cluster_var){
    counts_sc &lt;- as.matrix(logcounts(
      sce[, colData(sce)[, batch] %in% batch_var &amp; colData(sce)[, celltype] %in% cluster_var]))
  })
  mean_c &lt;- mean_cluster %&gt;% map(rowMeans) %&gt;% bind_rows %&gt;%
    dplyr::mutate(gene=rownames(sce)) %&gt;% gather(cluster, logcounts, levels(as.factor(colData(sce)[,celltype])))
})

mean_expr &lt;- mean_list %&gt;% bind_rows(.id= &quot;sample&quot;)

ggplot(mean_expr, aes(x=logcounts, colour=sample)) + geom_density(alpha=.3) +
  theme_classic() +
  facet_wrap( ~ cluster, ncol = 3) +
  scale_colour_manual(values = cols[c(1:3,7)]) +
  scale_x_continuous(limits =  c(0, 7))</code></pre>
<pre><code>Warning: Removed 4429 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/pancreas.Rmd/count%20distributions-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-count-distributions-1">
Past versions of “count distributions-1.png”
</button>
</p>
<div id="fig-no-spaces-count-distributions-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/count distributions-1.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/count distributions-1.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/count distributions-1.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/count distributions-1.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#number of cluster with differnt batch distributions
per_dist &lt;- 7</code></pre>
<p>Mean expression batch vs each other</p>
<pre class="r"><code>#mean expression
mean_expr &lt;- mean_list %&gt;% bind_rows(.id = &quot;sample&quot; ) %&gt;% spread(sample, logcounts)
batch_all &lt;- levels(as.factor(colData(sce)[,batch]))

lapply(batch_all, function(batch_var){
  batch_var_2 &lt;- batch_all[-which(batch_all %in% batch_var)]
  lapply(batch_var_2, function(batch_var_3){
    ggplot(mean_expr, aes_string(x=batch_var, y=batch_var_3)) +
      geom_point(alpha=.3, aes(color=cluster)) +
      ggtitle(batch_var) + geom_abline(slope = 1) + coord_fixed() +
      facet_wrap( ~ cluster, ncol = 3) +
      scale_color_manual(values=cols)
  })
})</code></pre>
<pre><code>[[1]]
[[1]][[1]]</code></pre>
<p><img src="figure/pancreas.Rmd/mean%20expression-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-mean-expression-1">
Past versions of “mean expression-1.png”
</button>
</p>
<div id="fig-no-spaces-mean-expression-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/mean expression-1.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/mean expression-1.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/mean expression-1.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[1]][[2]]</code></pre>
<p><img src="figure/pancreas.Rmd/mean%20expression-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-mean-expression-2">
Past versions of “mean expression-2.png”
</button>
</p>
<div id="fig-no-spaces-mean-expression-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/mean expression-2.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/mean expression-2.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/mean expression-2.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[2]]
[[2]][[1]]</code></pre>
<p><img src="figure/pancreas.Rmd/mean%20expression-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-mean-expression-3">
Past versions of “mean expression-3.png”
</button>
</p>
<div id="fig-no-spaces-mean-expression-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/mean expression-3.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/mean expression-3.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/mean expression-3.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]][[2]]</code></pre>
<p><img src="figure/pancreas.Rmd/mean%20expression-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-mean-expression-4">
Past versions of “mean expression-4.png”
</button>
</p>
<div id="fig-no-spaces-mean-expression-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/mean expression-4.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/mean expression-4.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/mean expression-4.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[3]]
[[3]][[1]]</code></pre>
<p><img src="figure/pancreas.Rmd/mean%20expression-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-mean-expression-5">
Past versions of “mean expression-5.png”
</button>
</p>
<div id="fig-no-spaces-mean-expression-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/mean expression-5.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/mean expression-5.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/mean expression-5.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]][[2]]</code></pre>
<p><img src="figure/pancreas.Rmd/mean%20expression-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-mean-expression-6">
Past versions of “mean expression-6.png”
</button>
</p>
<div id="fig-no-spaces-mean-expression-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/mean expression-6.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/mean expression-6.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/mean expression-6.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cellspecific-mixing-score" class="section level1">
<h1>Cellspecific mixing score</h1>
<div id="calculate-cms" class="section level2">
<h2>Calculate cms</h2>
<pre class="r"><code>sce &lt;- cms(sce, group = batch, k = 120, cell_min = 10, n_dim = 10)</code></pre>
<pre><code>Warning in .defineSubspace(sce, assay_name, dim_red, n_dim): &#39;dim_red&#39; not found:
            PCA subspace is used to calculate distances.</code></pre>
<pre><code>Warning in (function (jobs, data, centers, info, distance, k, get.index, :
tied distances detected in nearest-neighbor calculation</code></pre>
<pre class="r"><code>saveRDS(sce, paste0(out_path, &quot;/cms_&quot;, dataset_name, &quot;.rds&quot;))
sce &lt;- readRDS(paste0(out_path, &quot;/cms_&quot;, dataset_name, &quot;.rds&quot;))

#visualize
visHist(sce, n_col = 2)</code></pre>
<p><img src="figure/pancreas.Rmd/cms-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cms-1">
Past versions of cms-1.png
</button>
</p>
<div id="fig-cms-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/cms-1.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/cms-1.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/b6183ee8ddc6f60fa8c698d18838b90142058f4a/docs/figure/pancreas.Rmd/cms-1.png" target="_blank">b6183ee</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/e3b82df768b34dd6d1ee5042e77c9e026742352e/docs/figure/pancreas.Rmd/cms-1.png" target="_blank">e3b82df</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>visMetric(sce, metric_var = &quot;cms_smooth&quot;, dim_red = &quot;UMAP&quot;)</code></pre>
<p><img src="figure/pancreas.Rmd/cms-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cms-2">
Past versions of cms-2.png
</button>
</p>
<div id="fig-cms-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/c303fb7ac19b5809e1acf7fa2d0c8b937b1d77eb/docs/figure/pancreas.Rmd/cms-2.png" target="_blank">c303fb7</a>
</td>
<td>
almutlue
</td>
<td>
2019-07-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/f8e6fa112016bc7c89c55e0601b7a21d2e1e941c/docs/figure/pancreas.Rmd/cms-2.png" target="_blank">f8e6fa1</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/3729d1e6a0b865dcc057e24b1842f1376e353ff9/docs/figure/pancreas.Rmd/cms-2.png" target="_blank">3729d1e</a>
</td>
<td>
almutlue
</td>
<td>
2019-06-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/characterize_batches/blob/b6183ee8ddc6f60fa8c698d18838b90142058f4a/docs/figure/pancreas.Rmd/cms-2.png" target="_blank">b6183ee</a>
</td>
<td>
zjanna
</td>
<td>
2019-06-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>visGroup(sce, celltype, dim_red = &quot;UMAP&quot;)</code></pre>
<p><img src="figure/pancreas.Rmd/cms-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#summarize
mean_cms &lt;- mean(sce$cms)
n_cms_0.01 &lt;- length(which(sce$cms &lt; 0.01))
cluster_mean_cms &lt;- as_tibble(colData(sce)) %&gt;% group_by_at(celltype) %&gt;% summarize(cms_mean = mean(cms))
var_cms &lt;- var(cluster_mean_cms$cms_mean)</code></pre>
</div>
</div>
<div id="de-analysis" class="section level1">
<h1>DE analysis</h1>
<p>Calculate DEG</p>
<pre class="r"><code>x&lt;-sce
clust &lt;- as.factor(colData(sce)[,celltype])
# n_cells &lt;- table(clust,colData(sce)[,sample])
kids&lt;-levels(clust)
names(kids) &lt;- kids
group&lt;-as.factor(colData(sce)[,batch])

cs &lt;- names(cont)
names(cs) &lt;- cs
es&lt;- expr
ctype &lt;- &quot;contrast&quot;
res_df &lt;- function(k, tt, ct, c) {
  df &lt;- data.frame(
    gene = rownames(tt), cluster_id = k, tt,
    row.names = NULL, stringsAsFactors = FALSE)
  df[[ct]] &lt;- c
  return(df)
}
doDE &lt;- function(x,lfc_cutoff=log2(1.1)){
  res&lt;-lapply(kids, function (k) {
    cat(k, &quot;..&quot;, sep = &quot;&quot;)
    n &lt;- clust==k
    es_tmp &lt;- es[,n]
    grp &lt;- group[n]
    design &lt;- model.matrix(~0+grp)
    colnames(design)&lt;-levels(group)
    k1 &lt;- rowSums(es_tmp &gt; 0) &gt;= .2*min(table(grp))
    es_tmp &lt;- es_tmp[k1,]
    f &lt;- lmFit(es_tmp, design)
    f &lt;- eBayes(f, trend = TRUE)
    tt &lt;- lapply(cont, function(c) {
      cc&lt;-names(c)
      fc &lt;- contrasts.fit(f, contrasts = c)
      tr &lt;- treat(fc, lfc=lfc_cutoff)
      tt &lt;- topTreat(tr, n=Inf)
      res_df(k, tt, ctype,cc)
    })
    return(list(tt = tt, data = es_tmp))
  })
  # remove empty clusters
  skipped &lt;- vapply(res, is.null, logical(1))
  if (any(skipped))
    message(paste(&quot;Cluster(s)&quot;, dQuote(kids[skipped]), &quot;skipped due to an&quot;,
                  &quot;insufficient number of cells in at least 2 samples per group.&quot;))
  res &lt;- res[!skipped]
  kids &lt;- kids[names(res)]
  
  # re-organize by contrast &amp;
  # do global p-value adjustment
  tt &lt;- lapply(res, &quot;[[&quot;, &quot;tt&quot;)
  tt &lt;- lapply(cs, function(c) map(tt, c))
  
  # return results
  data &lt;- lapply(res, &quot;[[&quot;, &quot;data&quot;)
  list(table = tt,
       data = data,
       design = design,
       coef = coef)
}
res&lt;-doDE(x,lfc_cutoff=log2(1.1))</code></pre>
<pre><code>acinar..activated_stellate..alpha..beta..delta..ductal..endothelial..epsilon..</code></pre>
<pre><code>Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero</code></pre>
<pre><code>gamma..macrophage..</code></pre>
<pre><code>Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero</code></pre>
<pre><code>mast..</code></pre>
<pre><code>Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero</code></pre>
<pre><code>quiescent_stellate..</code></pre>
<pre><code>Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero</code></pre>
<pre><code>schwann..</code></pre>
<pre><code>Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero

Warning: Zero sample variances detected, have been offset away from zero</code></pre>
<pre class="r"><code>saveRDS(res, file=paste0(out_path, &quot;/limma_&quot;, dataset_name, &quot;.rds&quot;))</code></pre>
</div>
<div id="diagnostics-plot-of-deg" class="section level1">
<h1>Diagnostics plot of DEG</h1>
</div>
<div id="upset-plot" class="section level1">
<h1>Upset plot</h1>
<pre class="r"><code>res &lt;- readRDS(paste0(out_path, &quot;/limma_&quot;, dataset_name, &quot;.rds&quot;))

# Filter DEG by pvalue
FilterDEGs &lt;- function (degDF=df, filter=c(FDR=5))
{
  rownames(degDF)&lt;-degDF$gene
  pval &lt;- degDF[, grep(&quot;adj.P.Val$&quot;, colnames(degDF)), drop = FALSE]
  pf &lt;- pval &lt;= filter[&quot;FDR&quot;]/100
  pf[is.na(pf)] &lt;- FALSE
  DEGlistUPorDOWN &lt;- sapply(colnames(pf), function(x) rownames(pf[pf[, x, drop = FALSE], , drop = FALSE]), simplify = FALSE)
}

result &lt;- list()
m2 &lt;- list()

for(jj in 1:length(cs)){
  result[[jj]] &lt;- sapply(res[[1]][[names(cs)[jj]]], function(x) FilterDEGs(x))
  names(result[[jj]]) &lt;- kids
  m2[[jj]] = make_comb_mat(result[[jj]], mode = &quot;intersect&quot;)
}
names(result) &lt;- names(cs)
names(m2) &lt;- names(cs)

#lapply(m2, function(x) UpSet(x))</code></pre>
<div id="histograms-of-p-values" class="section level2">
<h2>Histograms of p-values</h2>
</div>
</div>
<div id="summarize-deg" class="section level1">
<h1>Summarize DEG</h1>
<pre class="r"><code># DE genes (per cluster and mean)
n_de &lt;- lapply(res[[1]],function(y) vapply(y, function(x) sum(x$adj.P.Val &lt; 0.05), numeric(1)))
n_de_cl &lt;- lapply(res[[1]],function(y) vapply(y, function(x) nrow(x), numeric(1)))
mean_n_de &lt;- lapply(n_de,function(x) mean(x))
mean_mean_n_de &lt;- mean(unlist(mean_n_de))/n_genes
min_mean_n_de &lt;- min(unlist(mean_n_de))/n_genes
max_mean_n_de &lt;- max(unlist(mean_n_de))/n_genes

# Genes with lfc &gt; 1
n_genes_lfc1 &lt;- lapply(res[[1]],function(y) vapply(y, function(x) sum(abs(x$logFC) &gt; 1), numeric(1)))
mean_n_genes_lfc1 &lt;- mean(unlist(n_genes_lfc1))/n_genes
min_n_genes_lfc1 &lt;- min(unlist(n_genes_lfc1))/n_genes
max_n_genes_lfc1 &lt;- max(unlist(n_genes_lfc1))/n_genes

# DE genes overlap between celltypes (celltype specific de genes)
# Genes are &quot;overlapping&quot; if they are present in all clusters with at least 10% of all cells
de_overlap &lt;- lapply(result,function(x){
  result2 &lt;- x[table(colData(sce)[, celltype]) &gt; n_cells * 0.1]
  de_overlap &lt;- length(Reduce(intersect, result2))
  de_overlap
})

mean_de_overlap &lt;- mean(unlist(de_overlap))/n_genes
min_de_overlap &lt;- min(unlist(de_overlap))/n_genes
max_de_overlap &lt;- max(unlist(de_overlap))/n_genes

#Genes unique to single celltypes
unique_genes_matrix &lt;- NULL
unique_genes &lt;- NULL
cb &lt;- length(names(result[[1]]))
unique_genes &lt;- lapply(result,function(x){
  for( i in 1:cb ){
    unique_genes[i] &lt;-as.numeric(length(setdiff(unlist(x[i]),unlist(x[-i]))))
  }
  unique_genes_matrix &lt;- cbind(unique_genes_matrix, unique_genes)
  unique_genes_matrix
})
unique_genes &lt;- Reduce(&#39;cbind&#39;, unique_genes)
colnames(unique_genes) &lt;- names(result)
rownames(unique_genes) &lt;- names(result[[1]])

# Relative cluster specificity (unique/overlapping)
rel_spec1 &lt;- NULL
for( i in 1:dim(unique_genes)[2] ){
  rel_spec &lt;- unique_genes[,i]/de_overlap[[i]]
  rel_spec1 &lt;- cbind(rel_spec1,rel_spec)
}

mean_rel_spec &lt;- mean(rel_spec1)
min_rel_spec &lt;- min(rel_spec1)
max_rel_spec &lt;- max(rel_spec1)</code></pre>
</div>
<div id="distribution-of-lfc-by-celltypes" class="section level1">
<h1>Distribution of lfc by celltypes</h1>
<p>To simulate a realistic variance by celltype we model the lfc by celltypes from the real underlying distribution. Most lfc follow a gamma distribution with varying parameter, but there seem to be a set of outstanding genes with a distinct higher lfc. To model them we use a second distribution for “type genes”. We need this to cover the maxima, but in general this second distribution does not seem to be sufficient representing ct changes either. So reality still seems to be more complex than classifying cells into 2 groups (“type” and “not type”).</p>
<pre class="r"><code># Determine lfc by celltype
block &lt;- colData(sce)[,batch]
kids &lt;- colData(sce)[,celltype]
res_ct &lt;- pairwiseTTests(
        assays(sce)$logcounts, 
        clusters = kids, block = block,
        direction = &quot;up&quot;, lfc = 0)
    
tbl &lt;- res_ct$statistics %&gt;% map_depth(1, data.frame)
names(tbl) &lt;- names(tbl) &lt;- res_ct$pairs$first

lfc_ct_all &lt;- tbl %&gt;% map(function(x){
    lfc &lt;- x$logFC
  }) %&gt;% unlist()

#variance within celltypes
lfc_by_k &lt;- tbl %&gt;% map(function(x){
    lfc &lt;- x$logFC
  }) %&gt;% bind_cols()

mean_lfc_tib &lt;- tibble(&quot;mean_lfc&quot; = colMeans(lfc_by_k), &quot;cluster&quot; = names(tbl)) %&gt;% 
  group_by(cluster) %&gt;% summarize(&quot;mean_k&quot; = mean(mean_lfc)) 

#distribution parameter 
mean_lfc_ct &lt;- mean(lfc_ct_all)
min_lfc_ct &lt;- min(lfc_ct_all)
max_lfc_ct &lt;- max(lfc_ct_all)

#Check lfc distribution
lfc_ct_real &lt;- data.frame(&quot;lfc&quot; = lfc_ct_all, &quot;type&quot; = rep(&quot;real&quot;, length(lfc_ct_all)))

###simulate overall celltype lfc distribution
#parameter
lfc_ct_shape &lt;- 0.8
lfc_ct_scale &lt;- 0.9

signs &lt;- sample(c(-1, 1), length(lfc_ct_all), TRUE, prob = c(0.5, 0.5))
tgam_ct &lt;- rtgamma(nrow(lfc_ct_real), lfc_ct_shape, lfc_ct_scale, a=min_lfc_ct, b=max_lfc_ct)*signs*0.5
lfc_sim_ct &lt;- data.frame(&quot;lfc&quot; = tgam_ct, &quot;type&quot; = rep(&quot;sim&quot;, nrow(lfc_ct_real)))


##### &quot;type genes&quot; distribution
type_gene &lt;- which(lfc_ct_all &gt; max(tgam_ct) | lfc_ct_all &lt; min(tgam_ct))
length(type_gene)</code></pre>
<pre><code>[1] 4366</code></pre>
<pre class="r"><code>lfc_type &lt;- data.frame(&quot;lfc&quot; = lfc_ct_all[type_gene], &quot;type&quot; = rep(&quot;type&quot;, length(type_gene)))

type_by_ct &lt;- colnames(lfc_by_k) %&gt;% map(function(x){
  v &lt;- lfc_by_k[,x]
  type &lt;- length(which(v &gt; max(tgam_ct) | v &lt; min(tgam_ct)))
}) %&gt;% unlist() %&gt;% set_names(names(tbl))

type_var &lt;- tibble(&quot;type_gene&quot; = type_by_ct, &quot;cluster&quot; = as.numeric(names(type_by_ct))) %&gt;% group_by(cluster) %&gt;% summarise(&quot;n_type&quot; = sum(type_gene)) %&gt;% mutate(&quot;p&quot; = n_type/(nrow(lfc_by_k) * ncol(lfc_by_k)/nrow(.)))</code></pre>
<pre><code>Warning in eval_tidy(xs[[i]], unique_output): NAs durch Umwandlung erzeugt</code></pre>
<pre class="r"><code>#Plot
lfc_comb &lt;- rbind(lfc_sim_ct, lfc_ct_real, lfc_type)
ggplot(lfc_comb, aes(lfc, colour = type)) + geom_density()</code></pre>
<p><img src="figure/pancreas.Rmd/lfc%20ct-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Simulation parameter
mean_lfc_type &lt;- mean(abs(lfc_ct_all[type_gene]))
sd_lfc_type &lt;- sd(abs(lfc_ct_all[type_gene]))
n_type &lt;- length(type_gene)/length(lfc_ct_all)</code></pre>
</div>
<div id="distribution-of-lfc-by-batches" class="section level1">
<h1>Distribution of lfc by batches</h1>
<p>Again we model 2 lfc distributions to cover the extremes as well.</p>
<pre class="r"><code>#logFoldchange batch effect distribution
mean_lfc_cl &lt;- lapply(res[[1]], function(y) vapply(y, function(x){
  de_genes &lt;- which(x$adj.P.Val &lt; 0.05)
  mean_de &lt;- mean(abs(x[, &quot;logFC&quot;]))}
  , numeric(1))) %&gt;% bind_cols()

min_lfc_cl &lt;- lapply(res[[1]],function(y) vapply(y, function(x){
  de_genes &lt;- which(x$adj.P.Val &lt; 0.05)
  min_de &lt;- min(x[, &quot;logFC&quot;])}
  , numeric(1))) %&gt;% bind_cols()

max_lfc_cl &lt;- lapply(res[[1]],function(y) vapply(y, function(x){
  de_genes &lt;- which(x$adj.P.Val &lt; 0.05)
  max_de &lt;- max(x[, &quot;logFC&quot;])}
  , numeric(1))) %&gt;% bind_cols()

mean_lfc_be &lt;- mean(colMeans(mean_lfc_cl, na.rm = TRUE))
min_lfc_be &lt;- min(colMins(as.matrix(min_lfc_cl), na.rm = TRUE))
max_lfc_be &lt;- max(colMaxs(as.matrix(max_lfc_cl), na.rm = TRUE))


#Check lfc distribution
lfc_dist &lt;- lapply(res[[1]], function(y){
  lapply(y, function(x){
  mean_de &lt;- x[, &quot;logFC&quot;]})
  }) %&gt;% unlist() %&gt;% as.data.frame() %&gt;% set_colnames(&quot;test&quot;) %&gt;% mutate(&quot;type&quot; = rep(&quot;real&quot;, nrow(.)))


#Simulation parameter
lfc_be_shape &lt;- 0.84
lfc_be_scale &lt;- 0.8

signs &lt;- sample(c(-1, 1), nrow(lfc_dist), TRUE, prob = c(0.7, 0.3))
tgam &lt;- rtgamma(nrow(lfc_dist), lfc_be_shape, scale=lfc_be_scale, a=min_lfc_be, b=max_lfc_be)*signs
lfc_sim &lt;- data.frame(&quot;test&quot; = tgam, &quot;type&quot; = rep(&quot;sim&quot;, nrow(lfc_dist)))

#&quot;batch genes&quot;
batch_gene &lt;- which(lfc_dist$test &gt; max(tgam) | lfc_dist$test &lt; min(tgam))
length(batch_gene)</code></pre>
<pre><code>[1] 193</code></pre>
<pre class="r"><code>lfc_batch &lt;- data.frame(&quot;test&quot; = lfc_dist$test[batch_gene], &quot;type&quot; = rep(&quot;batch&quot;, length(batch_gene)))

mean_lfc_batch &lt;- mean(abs(lfc_dist$test[batch_gene]))
sd_lfc_batch &lt;- sd(abs(lfc_dist$test[batch_gene]))
n_batch_genes &lt;- length(batch_gene)/nrow(lfc_dist)

signs &lt;- sample(c(-1, 1), length(batch_gene), TRUE, prob = c(0.1, 0.9))
tnorm &lt;- rnorm(length(batch_gene), mean = mean_lfc_batch, sd = sd_lfc_batch) * signs
lfc_batch_sim &lt;- data.frame(&quot;test&quot; = tnorm, &quot;type&quot; = rep(&quot;batch_sim&quot;, length(batch_gene)))

lfc_comb &lt;- rbind(lfc_sim, lfc_dist, lfc_batch, lfc_batch_sim)
ggplot(lfc_comb, aes(test, colour = type)) + geom_density()</code></pre>
<p><img src="figure/pancreas.Rmd/lfc%20batch-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="simulation-parameter" class="section level1">
<h1>Simulation parameter</h1>
<p>Define parameter to describe the dataset for simulation</p>
<pre class="r"><code>#percentage of batch affected genes
cond &lt;- gsub(&quot;-.*&quot;, &quot;&quot;, names(n_de))
cond &lt;- c(cond, unique(gsub(&quot;.*-&quot;, &quot;&quot;, names(n_de))))
cond &lt;- unique(cond)
de_be_tab &lt;- n_de %&gt;% bind_cols()
de_cl_tab &lt;- n_de_cl %&gt;% bind_cols()
de_be &lt;- cond %&gt;% map(function(x){
  de_tab &lt;- de_be_tab[, grep(x, colnames(de_be_tab))]
  de_be &lt;- rowMeans(de_tab)
}) %&gt;% bind_cols() %&gt;% set_colnames(cond)

n_cl &lt;- cond %&gt;% map(function(x){
  cl_tab &lt;- de_cl_tab[, grep(x, colnames(de_cl_tab))]
  de_cl &lt;- rowMeans(cl_tab)
}) %&gt;% bind_cols() %&gt;% set_colnames(cond)


p_be &lt;- de_be/n_cl
mean_p_be &lt;- mean(colMeans(p_be))
min_p_be &lt;- min(colMins(as.matrix(p_be)))
max_p_be &lt;- max(colMaxs(as.matrix(p_be)))
sd_p_be &lt;- mean(colSds(as.matrix(p_be)))
if(is.na(sd_p_be)){ sd_p_be &lt;- 0 }

## Could also be inferred from the variance explained by the batches? (also arbitrary threshold)
p_be_mod &lt;- n_batch_gene_mod


#### How much does the batch fold change vary between celltypes? &quot;rel_be&quot;
var_lfc_cl &lt;- colSds(as.matrix(mean_lfc_cl), na.rm = TRUE)
rel_be &lt;- mean(var_lfc_cl)

#### How much does the celltype fold change vary between celltypes? &quot;rel_ct&quot;
rel_ct &lt;- sd(mean_lfc_tib$mean_k)

#### Percentage of celltype specific genes &quot;p_ct&quot;
n_de_unique &lt;- lapply(result,function(x){
  de_genes &lt;- unlist(x) %&gt;% unique() %&gt;% length()
  de_genes &lt;- de_genes/length(x)
}) %&gt;% bind_cols()


rel_spec2 &lt;- NULL
for(i in 1:length(de_overlap)){
  rel_spec &lt;- de_overlap[[i]]/mean(n_de[[i]][table(colData(sce)[,celltype])&gt;dim(expr)[2]*0.1])
  rel_spec2 &lt;- cbind(rel_spec2,rel_spec)
}

mean_p_ct &lt;- 1 - mean(rel_spec2)
max_p_ct &lt;- 1 - min(rel_spec2)
min_p_ct &lt;- 1 - max(rel_spec2)
sd_p_ct &lt;- sd(rel_spec2)
if(is.na(sd_p_ct)){ sd_p_ct &lt;- 0 }</code></pre>
</div>
<div id="summarize-results" class="section level1">
<h1>Summarize results</h1>
<pre class="r"><code>#Size? How much of the variance can be attributed to the batch effect?
size &lt;- data.frame(&quot;batch_genes_1per&quot; = n_batch_gene,
                   &quot;batch_genes_10per&quot; = n_batch_gene10,
                   &quot;celltype_gene_1per&quot; = n_celltype_gene,
                   &quot;relative_batch_celltype&quot; = n_rel,
                   &quot;batch_genes_1per_mod&quot; = n_batch_gene_mod,
                   &quot;batch_genes_10per_mod&quot; = n_batch_gene10_mod,
                   &quot;celltype_gene_1per_mod&quot; = n_celltype_gene_mod,
                   &quot;relative_batch_celltype_mod&quot; = n_rel_mod,
                   &quot;mean_var_batch&quot; = m_batch,
                   &quot;mean_var_celltype&quot; = m_celltype,
                   &quot;median_var_batch&quot; = me_batch,
                   &quot;median_var_celltype&quot; = me_celltype,
                   &quot;mean_var_batch_mod&quot; = m_batch_mod,
                   &quot;mean_var_celltype_mod&quot; = m_celltype_mod,
                   &quot;median_var_batch_mod&quot; = me_batch_mod,
                   &quot;median_var_celltype_mod&quot; = me_celltype_mod,
                   &quot;n_cells_total&quot; = ncol(sce),
                   &quot;n_genes_total&quot; = nrow(sce))


#Celltype-specificity? How celltype/cluster specific are batch effects? Differences in sample variation between batches?
celltype &lt;- data.frame(&quot;per_count_dist&quot; = per_dist,
                       &quot;mean_cms&quot; = mean_cms,
                       &#39;mean_rel_abund_diff&#39; = mean_rel_abund_diff,
                       &#39;min_rel_abund_diff&#39; = min_rel_abund_diff,
                       &#39;max_rel_abund_diff&#39; = max_rel_abund_diff,
                       &quot;celltype_var_cms&quot; = var_cms,
                       &quot;n_cells_cms_0.01&quot; = n_cms_0.01)

#Gene-specificity? How do they effect genes? Single genes? All genes? Which genes?
gene &lt;- data.frame(&quot;mean_mean_n_de_genes&quot; = mean_mean_n_de,
                   &quot;max_mean_n_de_genes&quot; = max_mean_n_de,
                   &quot;min_mean_n_de_genes&quot; = min_mean_n_de,
                   &quot;mean_n_genes_lfc1&quot; = mean_n_genes_lfc1,
                   &quot;min_n_genes_lfc1&quot; = min_n_genes_lfc1,
                   &quot;max_n_genes_lfc1&quot; = max_n_genes_lfc1,
                   &quot;mean_de_overlap&quot; = mean_de_overlap,
                   &quot;min_de_overlap&quot; = min_de_overlap,
                   &quot;max_de_overlap&quot; = max_de_overlap,
                   &quot;mean_rel_cluster_spec&quot;= mean_rel_spec,
                   &quot;min_rel_cluster_spec&quot;= min_rel_spec,
                   &quot;max_rel_cluster_spec&quot;= max_rel_spec
)
# Cell-specificity? How cell-specific are batche effects? Are their differences in within celltype variation between batches?

sim &lt;- data.frame(&quot;mean_p_be&quot; = mean_p_be,
                  &quot;max_p_be&quot; = max_p_be,
                  &quot;min_p_be&quot; = min_p_be,
                  &quot;sd_p_be&quot; = sd_p_be,
                  &quot;p_be_mod&quot; = p_be_mod,
                  &quot;mean_lfc_be&quot; = mean_lfc_be,
                  &quot;min_lfc_be&quot; = min_lfc_be,
                  &quot;max_lfc_be&quot; = max_lfc_be,
                  &quot;mean_lfc_ct&quot; = mean_lfc_ct,
                  &quot;min_lfc_ct&quot; = min_lfc_ct,
                  &quot;max_lfc_ct&quot; = max_lfc_ct,
                  &quot;mean_rel_be&quot; = rel_be,
                  &quot;rel_ct&quot; = rel_ct,
                  &quot;mean_p_ct&quot;= mean_p_ct,
                  &quot;min_p_ct&quot;= min_p_ct,
                  &quot;max_p_ct&quot;= max_p_ct,
                  &quot;sd_p_ct&quot; = sd_p_ct,
                  &quot;mean_lfc_batch&quot; = mean_lfc_batch,
                  &quot;sd_lfc_batch&quot; = 0,
                  &quot;n_batch_gene&quot; = n_batch_genes,
                  &quot;mean_lfc_type&quot; = mean_lfc_type,
                  &quot;sd_lfc_type&quot; = sd_lfc_type,
                  &quot;n_type&quot; = n_type,
                  &quot;lfc_ct_shape&quot; = lfc_ct_shape,
                  &quot;lfc_ct_scale&quot; = lfc_ct_scale,
                  &quot;lfc_be_shape&quot; = lfc_be_shape,
                  &quot;lfc_be_scale&quot; = lfc_be_scale)

summary &lt;- cbind(size, celltype, gene, sim) %&gt;% set_rownames(dataset_name)
saveRDS(summary, paste0(out_path, &quot;/summary_&quot;, dataset_name, &quot;.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=de_DE.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=de_DE.UTF-8    
 [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=de_DE.UTF-8   
 [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] scran_1.12.1                TruncatedDistributions_1.0 
 [3] ggtern_3.1.0                variancePartition_1.14.0   
 [5] scales_1.0.0                foreach_1.4.4              
 [7] limma_3.40.2                ComplexHeatmap_2.1.0       
 [9] diffcyt_1.4.3               stringr_1.4.0              
[11] dplyr_0.8.3                 tidyr_0.8.3                
[13] here_0.1                    purrr_0.3.2                
[15] gridExtra_2.3               CellMixS_1.1.0             
[17] kSamples_1.2-9              SuppDists_1.1-9.4          
[19] jcolors_0.0.4               scater_1.12.2              
[21] ggplot2_3.2.0               CellBench_1.0.0            
[23] tibble_2.1.3                magrittr_1.5               
[25] SingleCellExperiment_1.6.0  SummarizedExperiment_1.14.0
[27] DelayedArray_0.10.0         BiocParallel_1.18.0        
[29] matrixStats_0.54.0          Biobase_2.44.0             
[31] GenomicRanges_1.36.0        GenomeInfoDb_1.20.0        
[33] IRanges_2.18.1              S4Vectors_0.22.0           
[35] BiocGenerics_0.30.0        

loaded via a namespace (and not attached):
  [1] proto_1.0.0                 tidyselect_0.2.5           
  [3] lme4_1.1-21                 RSQLite_2.1.1              
  [5] flowCore_1.50.0             munsell_0.5.0              
  [7] codetools_0.2-16            statmod_1.4.32             
  [9] withr_2.1.2                 colorspace_1.4-1           
 [11] energy_1.7-6                highr_0.8                  
 [13] knitr_1.23                  robustbase_0.93-5          
 [15] bayesm_3.1-2                labeling_0.3               
 [17] git2r_0.26.1                GenomeInfoDbData_1.2.1     
 [19] bit64_0.9-7                 rprojroot_1.3-2            
 [21] vctrs_0.2.0                 TH.data_1.0-10             
 [23] xfun_0.8                    BiocFileCache_1.8.0        
 [25] R6_2.4.0                    doParallel_1.0.14          
 [27] ggbeeswarm_0.6.0            clue_0.3-57                
 [29] rsvd_1.0.1                  locfit_1.5-9.1             
 [31] bitops_1.0-6                assertthat_0.2.1           
 [33] multcomp_1.4-10             beeswarm_0.2.3             
 [35] gtable_0.3.0                sandwich_2.5-1             
 [37] workflowr_1.4.0             rlang_0.4.0                
 [39] zeallot_0.1.0               GlobalOptions_0.1.0        
 [41] splines_3.6.1               lazyeval_0.2.2             
 [43] yaml_2.2.0                  reshape2_1.4.3             
 [45] backports_1.1.4             tensorA_0.36.1             
 [47] tools_3.6.1                 gplots_3.0.1.1             
 [49] RColorBrewer_1.1-2          dynamicTreeCut_1.63-1      
 [51] ggridges_0.5.1              latex2exp_0.4.0            
 [53] Rcpp_1.0.1                  plyr_1.8.4                 
 [55] progress_1.2.2              zlibbioc_1.30.0            
 [57] RCurl_1.95-4.12             FlowSOM_1.16.0             
 [59] prettyunits_1.0.2           GetoptLong_0.1.7           
 [61] viridis_0.5.1               cowplot_1.0.0              
 [63] zoo_1.8-6                   cluster_2.1.0              
 [65] colorRamps_2.3              fs_1.3.1                   
 [67] RSpectra_0.15-0             circlize_0.4.6             
 [69] mvtnorm_1.0-11              whisker_0.3-2              
 [71] hms_0.5.0                   evaluate_0.14              
 [73] pbkrtest_0.4-7              XML_3.98-1.20              
 [75] shape_1.4.4                 compiler_3.6.1             
 [77] KernSmooth_2.23-15          crayon_1.3.4               
 [79] minqa_1.2.4                 htmltools_0.3.6            
 [81] corpcor_1.6.9               pcaPP_1.9-73               
 [83] rrcov_1.4-7                 RcppParallel_4.4.3         
 [85] lubridate_1.7.4             DBI_1.0.0                  
 [87] dbplyr_1.4.2                MASS_7.3-51.4              
 [89] rappdirs_0.3.1              boot_1.3-23                
 [91] compositions_1.40-2         Matrix_1.2-17              
 [93] gdata_2.18.0                igraph_1.2.4.1             
 [95] pkgconfig_2.0.2             vipor_0.4.5                
 [97] dqrng_0.2.1                 XVector_0.24.0             
 [99] digest_0.6.20               RcppAnnoy_0.0.12           
[101] tsne_0.1-3                  ConsensusClusterPlus_1.48.0
[103] graph_1.62.0                rmarkdown_1.14             
[105] uwot_0.1.3                  edgeR_3.26.5               
[107] DelayedMatrixStats_1.6.0    listarrays_0.2.0           
[109] curl_3.3                    gtools_3.8.1               
[111] rjson_0.2.20                nloptr_1.2.1               
[113] nlme_3.1-140                BiocNeighbors_1.2.0        
[115] viridisLite_0.3.0           pillar_1.4.2               
[117] lattice_0.20-38             httr_1.4.0                 
[119] DEoptimR_1.0-8              survival_2.44-1.1          
[121] glue_1.3.1                  png_0.1-7                  
[123] iterators_1.0.10            bit_1.1-14                 
[125] stringi_1.4.3               blob_1.2.0                 
[127] BiocSingular_1.0.0          caTools_1.17.1.2           
[129] memoise_1.1.0               irlba_2.3.3                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
