<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="almutlue" />

<meta name="date" content="2019-05-28" />

<title>pbmc_media_storage</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">characterize_batches</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">pbmc_media_storage</h1>
<h4 class="author">almutlue</h4>
<h4 class="date">2019-05-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-05-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>characterize_batches/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.3.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190528code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190528)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190528code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190528)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong8b934db"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 8b934db </a>
</p>
</div>
<div id="strongRepositoryversionstrong8b934db" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    docs/figure/

Untracked files:
    Untracked:  data/pbmc_media.rds
    Untracked:  output/pbmc_media_cms.rds
    Untracked:  output/summary_pbmc_media_storage.rds
    Untracked:  output/vp_pbmc.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
8b934db
</td>
<td>
almutlue
</td>
<td>
2019-05-28
</td>
<td>
Add pbmc analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="different-media-conditions" class="section level1">
<h1>Different media conditions</h1>
<p>Pbmc - media storage</p>
<p><strong>PBMC dataset</strong> from Roche that has been collected to test the influence of media storage on sequencing. It includes 11045 human PBMCs from the same sample, split into 4 batches. All batches have been sequenced together and libraries have been prepared together, but 3 batches have been stored in different media DMSO, CSF and PSC and frozen for 7 days before sequencing. The last batch contains PBMCs from freshly collected blood. Differences between these batches should all derive from media storage, so conditional sources.</p>
<pre class="r"><code>suppressPackageStartupMessages({
  library(dplyr)
  library(CellMixS)
  library(scater)
  library(scran)
  library(variancePartition)
  library(here)
  library(jcolors)
  library(magrittr)
  library(purrr)
  library(diffcyt)
  library(tidyr)
})</code></pre>
<pre class="r"><code># data 
data_path &lt;- here::here(&quot;data&quot;)
out_path &lt;- here::here(&quot;output&quot;)

# raw data 
#For preprocessing and quality control please check: /code/pbmc_media.rmd
sce &lt;- readRDS(paste0(data_path, &quot;/pbmc_media.rds&quot;))

cols &lt;-c(c(jcolors(&#39;pal6&#39;),jcolors(&#39;pal8&#39;))[c(1,8,14,5,2:4,6,7,9:13,15:20)],jcolors(&#39;pal4&#39;))
names(cols) &lt;- c()</code></pre>
</div>
<div id="visualize-batch-effect" class="section level1">
<h1>Visualize batch effect</h1>
<pre class="r"><code>feature_list &lt;- c(&quot;sample_type&quot;, &quot;cluster.merged&quot;, &quot;cluster.main&quot;)

lapply(feature_list, function(feature_name){
  visGroup(sce, feature_name, dim_red= &quot;tsne&quot;)
})</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/vis%20batch-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/vis%20batch-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/vis%20batch-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sizestrength-of-the-batch-effcet" class="section level1">
<h1>Size/Strength of the batch effcet</h1>
<p>How much of the variance within data can be attributed to the batch effect?</p>
<div id="variancepartitioning" class="section level2">
<h2>VariancePartitioning</h2>
<pre class="r"><code>expr &lt;- as.matrix(assays(sce)$logcounts)
meta_sub &lt;- as.data.frame(colData(sce)[, c(&quot;cluster.merged&quot;, &quot;dataset&quot;)])
form &lt;- ~ (1|cluster.merged) + (1|dataset)

#varPart &lt;- fitExtractVarPartModel(expr, form, meta_sub) 

#Sort variables (i.e. columns) by median fraction# of variance explained
#vp &lt;- sortCols(varPart)
vp &lt;- readRDS(paste0(out_path,&quot;/vp_pbmc.rds&quot;))
vp_names &lt;- rownames(vp)
vp &lt;-vp  %&gt;% dplyr::mutate(gene= vp_names) %&gt;% dplyr::arrange(-dataset)</code></pre>
<pre><code>Warning in class(x) &lt;- c(subclass, tibble_class): Setze class(x) auf
mehrere Zeichenketten(&quot;tbl_df&quot;, &quot;tbl&quot;, ...); das Ergebnis ist kein S4
Objekt mehr</code></pre>
<pre class="r"><code>vp_sub &lt;- vp[1:3] %&gt;% set_rownames(vp$gene)

#plot 
plotPercentBars( vp_sub[1:10,] )</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/variance%20part-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotVarPart( vp_sub )</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/variance%20part-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Summarize variance partitioning</p>
<pre class="r"><code>#How many genes have a variance component affected by the batch variable with more than 1%
n_batch_gene &lt;- as_tibble(vp) %&gt;%  dplyr::filter(dataset &gt; 0.01) %&gt;% nrow()
n_batch_gene10 &lt;- as_tibble(vp) %&gt;%  dplyr::filter(dataset &gt; 0.1) %&gt;% nrow()
n_celltype_gene &lt;- as_tibble(vp) %&gt;%  dplyr::filter(cluster.merged&gt; 0.01) %&gt;% nrow()
n_rel &lt;- n_batch_gene/n_celltype_gene

#The mean percentage of the variance that is explained by the batch effect independent from the celltype
m_batch &lt;- mean(vp$dataset)
m_celltype &lt;- mean(vp$cluster.merged)
m_rel &lt;- m_batch/m_celltype


#The median percentage of the variance that is explained by the batch effect independent from the celltype
me_batch &lt;- median(vp$dataset)
me_celltype &lt;- median(vp$cluster.merged)
me_rel &lt;- me_batch/me_celltype</code></pre>
</div>
</div>
<div id="celltype-specificity" class="section level1">
<h1>Celltype specificity</h1>
<div id="celltype-abundance" class="section level2">
<h2>Celltype abundance</h2>
<pre class="r"><code>meta_tib &lt;- as_tibble(colData(sce)) %&gt;% group_by(sample_type, cluster.merged) %&gt;% summarize(n = n()) %&gt;% dplyr::mutate(cell_freq = n / sum(n))


plot_abundance &lt;- function(cluster_var, tib, x_var){
  meta_df &lt;- as.data.frame(eval(tib))
  p &lt;-ggplot(data=meta_df, aes_string(x=x_var, y=&quot;cell_freq&quot;, fill = cluster_var)) +
  geom_bar(stat=&quot;identity&quot;) + scale_fill_manual(values=cols)
  p + coord_flip() + theme_minimal()
}

plot_abundance(cluster_var = &quot;cluster.merged&quot;, tib = meta_tib, x_var = &quot;sample_type&quot;)</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/celltype%20abundance-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Test for significant changes in celltype abundances. Diffcyt (skip for datasets without replicates)</p>
<pre class="r"><code>#colData
colData(sce)$sample_id &lt;- colData(sce)$dataset
sce$media &lt;- ifelse(grepl(&quot;Fresh PBMCs&quot;, sce$sample_type), &quot;fresh&quot;, &quot;frozen&quot;)

col_dat &lt;-  data.frame(&quot;sample_id&quot; = levels(sce$sample_id), &quot;group_id&quot; = c(&quot;fresh&quot;, rep(&quot;frozen&quot;, 3))) %&gt;% set_rownames(levels(sce$sample_id)) 


#create desing matrix
design_ &lt;- createDesignMatrix(
  col_dat, cols_design = c(&quot;group_id&quot;)
)

#create contrast
contrast &lt;- createContrast(c(0, 1))

#create summarizedExperiment for cluster abundance (rows = cluster, columns= samples)
cluster_counts &lt;- as_tibble(colData(sce)) %&gt;% group_by(cluster.merged, sample_id) %&gt;% summarize(n=n()) %&gt;%  spread(sample_id, n) 
cluster_count &lt;- as.matrix(cluster_counts[,2:5])
rownames(cluster_count) &lt;- cluster_counts$cluster.merged

#rowData
row_dat &lt;- as_tibble(colData(sce)) %&gt;% group_by(cluster.merged) %&gt;% summarize(n=n()) %&gt;% set_colnames(c(&quot;cluster_id&quot;, &quot;n_cells&quot;))



se &lt;- SummarizedExperiment(list(&quot;counts&quot; = cluster_count), rowData= as.data.frame(row_dat), colData= col_dat)


#differential abundance
res_DA &lt;- testDA_edgeR(se, design_, contrast)

#summarize - number of differential abundance cluster
n_da_cluster &lt;- length(which(rowData(res_DA)$p_adj &lt; 0.1))</code></pre>
</div>
</div>
<div id="count-distribution" class="section level1">
<h1>Count distribution</h1>
<p>Do the overall counts distribution vary between batches? Clusterwise ploting?</p>
<pre class="r"><code>#sample an cluster ids
sids &lt;- levels(as.factor(sce$sample_type))
names(sids) &lt;- sids
cids &lt;- levels(as.factor(sce$cluster.merged))
names(cids) &lt;- cids

#mean gene expression by sample and cluster
mean_list &lt;- lapply(sids, function(batch){
  mean_cluster &lt;- lapply(cids, function(cluster){
    counts_sc &lt;- as.matrix(logcounts(
        sce[,sce$sample_type %in% batch &amp; sce$cluster.merged %in% cluster]))
  })
  mean_c &lt;- mean_cluster %&gt;% map(rowMeans) %&gt;% bind_rows %&gt;%
    dplyr::mutate(gene=rownames(sce)) %&gt;% gather(cluster, counts, &quot;0&quot;:&quot;6&quot;) 
  })

mean_expr &lt;- mean_list %&gt;% bind_rows(.id= &quot;sample&quot;) 

ggplot(mean_expr, aes(x=counts, colour=sample)) + geom_density(alpha=.3) + 
    theme_classic() +
    facet_wrap( ~ cluster, ncol = 3) +
    scale_colour_manual(values = cols[c(1:3,7)]) +
    scale_x_continuous(limits =  c(0, 0.4))</code></pre>
<pre><code>Warning: Removed 27892 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/count%20distributions-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#number of cluster with differnt batch distributions
n_dist &lt;- 1</code></pre>
<p>Mean expression counts media vs fresh</p>
<pre class="r"><code>#mean expression
sids &lt;- levels(as.factor(sce$sample_type))
names(sids) &lt;- sids
cids &lt;- levels(as.factor(sce$cluster.merged))
names(cids) &lt;- cids
  
mean_list &lt;- lapply(sids, function(batch){
      mean_cluster &lt;- lapply(cids, function(cluster){
        counts_sc &lt;- as.matrix(logcounts(
          sce[,sce$sample_type %in% batch &amp; sce$cluster.merged %in% cluster]))
      })
    mean_c &lt;- mean_cluster %&gt;% map(rowMeans) %&gt;% bind_rows %&gt;% dplyr::mutate(gene=rownames(sce)) %&gt;% gather(cluster, counts, &quot;0&quot;:&quot;6&quot;) 
  })
  
  mean_expr &lt;- mean_list %&gt;% bind_rows(.id = &quot;sample&quot; ) %&gt;% spread(sample, counts)
  colnames(mean_expr) &lt;- c(&quot;gene&quot;, &quot;cluster&quot;, &quot;fresh&quot;, &quot;CS10&quot;, &quot;DMSO&quot;, &quot;PSC7&quot;)
  
  lapply(c(&quot;CS10&quot;, &quot;DMSO&quot;, &quot;PSC7&quot;), function(media){
    ggplot(mean_expr, aes_string(x=&quot;fresh&quot;, y=media)) + 
      geom_point(alpha=.3, aes(color=cluster)) + 
    ggtitle(media) + geom_abline(slope = 1) + coord_fixed() +
    facet_wrap( ~ cluster, ncol = 3) +
    scale_color_manual(values=cols)
  })</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/mean%20expression-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/mean%20expression-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/mean%20expression-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cellspecific-mixing-score" class="section level1">
<h1>Cellspecific mixing score</h1>
<p>Calculate cms</p>
<pre class="r"><code>#sce &lt;- cms(sce, group = &quot;sample_type&quot;, k = 200, k_min = 60, cell_min = 20)
#saveRDS(sce, paste0(out_path, &quot;/pbmc_media_cms.rds&quot;))
sce &lt;- readRDS(paste0(out_path, &quot;/pbmc_media_cms.rds&quot;))

visHist(sce, n_col = 2)</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/cms-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>visMetric(sce, metric_var = &quot;cms_smooth&quot;, dim_red = &quot;tsne&quot;)</code></pre>
<p><img src="figure/pbmc_media_storage.Rmd/cms-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#summarize
mean_cms &lt;- mean(sce$cms)
n_cms_0.01 &lt;- length(which(sce$cms &lt; 0.01))
cluster_mean_cms &lt;- as_tibble(colData(sce)) %&gt;% group_by(cluster.merged) %&gt;% summarize(cms_mean = mean(cms))

var_cms &lt;- sd(cluster_mean_cms$cms_mean)</code></pre>
</div>
<div id="summarize-results" class="section level1">
<h1>Summarize results</h1>
<pre class="r"><code>#Size? How much of the variance can be attributed to the batch effect?
size &lt;- data.frame(&quot;batch_genes_1per&quot; = n_batch_gene,  
                   &quot;batch_genes_10per&quot; = n_batch_gene10,
                   &quot;celltype_gene_1per&quot; = n_celltype_gene,
                   &quot;relative_batch_celltype&quot; = n_rel,
                   &quot;mean_var_batch&quot; = m_batch,
                   &quot;mean_var_celltype&quot; = m_celltype,
                   &quot;median_var_batch&quot; = me_batch,
                   &quot;median_var_celltype&quot; = me_celltype)


#Celltype-specificity? How celltype/cluster specific are batch effects? Differences in sample variation between batches?
celltype &lt;- data.frame(&quot;DA_celltypes&quot; = n_da_cluster,
                       &quot;n_count_dist&quot; = n_dist,
                       &quot;mean_cms&quot; = mean_cms,
                       &quot;celltype_var_cms&quot; = var_cms,
                       &quot;n_cells_cms_0.01&quot; = n_cms_0.01)

#Gene-specificity? How do they effect genes? Single genes? All genes? Which genes?
#gene &lt;- data.frame(&quot;mean_n_de_genes&quot; = mean_n_de,
#                   &quot;n_genes_lfc2&quot; = n_genes_lfc2,
#                   &quot;DE_overlap&quot; = de_overlap,
#                   &quot;DE_overlap50&quot; = de_overlap50,
#                   &quot;rel_cluster_specific_DE&quot; = rel_spec
#                   )
# Cell-specificity? How cell-specific are batche effects? Are their differences in within celltype variation between batches?

summary &lt;- cbind(size, celltype )#, gene) %&gt;% set_rownames(&quot;pbmc_media_storage&quot;)
saveRDS(summary, paste0(out_path, &quot;/summary_pbmc_media_storage.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=de_DE.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=de_DE.UTF-8    
 [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=de_DE.UTF-8   
 [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] tidyr_0.8.3                 diffcyt_1.4.3              
 [3] purrr_0.3.2                 magrittr_1.5               
 [5] jcolors_0.0.4               here_0.1                   
 [7] variancePartition_1.14.0    scales_1.0.0               
 [9] foreach_1.4.4               limma_3.40.2               
[11] scran_1.12.0                scater_1.12.2              
[13] ggplot2_3.1.1               SingleCellExperiment_1.6.0 
[15] SummarizedExperiment_1.14.0 DelayedArray_0.10.0        
[17] BiocParallel_1.18.0         matrixStats_0.54.0         
[19] Biobase_2.44.0              GenomicRanges_1.36.0       
[21] GenomeInfoDb_1.20.0         IRanges_2.18.0             
[23] S4Vectors_0.22.0            BiocGenerics_0.30.0        
[25] CellMixS_1.0.0              kSamples_1.2-9             
[27] SuppDists_1.1-9.4           dplyr_0.8.1                

loaded via a namespace (and not attached):
  [1] TH.data_1.0-10              ggbeeswarm_0.6.0           
  [3] minqa_1.2.4                 colorspace_1.4-1           
  [5] rjson_0.2.20                colorRamps_2.3             
  [7] ggridges_0.5.1              dynamicTreeCut_1.63-1      
  [9] rprojroot_1.3-2             circlize_0.4.6             
 [11] corpcor_1.6.9               XVector_0.24.0             
 [13] GlobalOptions_0.1.0         BiocNeighbors_1.2.0        
 [15] fs_1.3.1                    clue_0.3-57                
 [17] mvtnorm_1.0-10              codetools_0.2-16           
 [19] splines_3.6.0               doParallel_1.0.14          
 [21] robustbase_0.93-5           knitr_1.23                 
 [23] workflowr_1.3.0             nloptr_1.2.1               
 [25] pbkrtest_0.4-7              cluster_2.0.9              
 [27] png_0.1-7                   graph_1.62.0               
 [29] rrcov_1.4-7                 compiler_3.6.0             
 [31] dqrng_0.2.1                 backports_1.1.4            
 [33] assertthat_0.2.1            Matrix_1.2-17              
 [35] lazyeval_0.2.2              BiocSingular_1.0.0         
 [37] htmltools_0.3.6             prettyunits_1.0.2          
 [39] tools_3.6.0                 rsvd_1.0.0                 
 [41] igraph_1.2.4.1              gtable_0.3.0               
 [43] glue_1.3.1                  GenomeInfoDbData_1.2.1     
 [45] reshape2_1.4.3              Rcpp_1.0.1                 
 [47] gdata_2.18.0                nlme_3.1-140               
 [49] iterators_1.0.10            DelayedMatrixStats_1.6.0   
 [51] xfun_0.7                    stringr_1.4.0              
 [53] lme4_1.1-21                 irlba_2.3.3                
 [55] gtools_3.8.1                XML_3.98-1.19              
 [57] statmod_1.4.30              DEoptimR_1.0-8             
 [59] edgeR_3.26.3                zoo_1.8-5                  
 [61] zlibbioc_1.30.0             MASS_7.3-51.4              
 [63] listarrays_0.2.0            hms_0.4.2                  
 [65] sandwich_2.5-1              RColorBrewer_1.1-2         
 [67] ComplexHeatmap_2.0.0        yaml_2.2.0                 
 [69] gridExtra_2.3               stringi_1.4.3              
 [71] pcaPP_1.9-73                flowCore_1.50.0            
 [73] caTools_1.17.1.2            boot_1.3-22                
 [75] shape_1.4.4                 rlang_0.3.4                
 [77] pkgconfig_2.0.2             bitops_1.0-6               
 [79] evaluate_0.13               lattice_0.20-38            
 [81] labeling_0.3                cowplot_0.9.4              
 [83] tidyselect_0.2.5            plyr_1.8.4                 
 [85] R6_2.4.0                    gplots_3.0.1.1             
 [87] multcomp_1.4-10             pillar_1.4.0               
 [89] whisker_0.3-2               withr_2.1.2                
 [91] survival_2.44-1.1           RCurl_1.95-4.12            
 [93] tsne_0.1-3                  FlowSOM_1.16.0             
 [95] tibble_2.1.1                crayon_1.3.4               
 [97] KernSmooth_2.23-15          rmarkdown_1.13             
 [99] viridis_0.5.1               GetoptLong_0.1.7           
[101] progress_1.2.2              locfit_1.5-9.1             
[103] grid_3.6.0                  git2r_0.25.2               
[105] ConsensusClusterPlus_1.48.0 digest_0.6.19              
[107] munsell_0.5.0               beeswarm_0.2.3             
[109] viridisLite_0.3.0           vipor_0.4.5                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
