---
title: "cellbench_seq_protocols"
author: "almutlue"
date: "2019-05-29"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Technical batch effects
**Cellbench data**, test batch effect from different protocols. Using the same three celllines sequenced by sequencing methods.
This represents a technical batch effect.

libraries
```{r libs}
suppressPackageStartupMessages({
  library(CellBench)
  library(scater)
  library(jcolors)
  library(CellMixS)
  library(gridExtra)
  library(purrr)
  library(jcolors)
  library(here)
  library(tidyr)
  library(dplyr)
  library(stringr)
  library(variancePartition)
  library(diffcyt)
  library(ComplexHeatmap)
})
```

Dataset
```{r load data}
# data
out_path <- here::here("output")
code_path <- here::here("code")
dataset_name <- "cellBench"

#Mixology dataset
sc_data <- load_sc_data()
colData(sc_data[[1]])$protocol <- rep(names(sc_data)[1], ncol(sc_data[[1]]))
sce <- sc_data[[1]]

for(i in 2:length(sc_data)){
  colData(sc_data[[i]])$protocol <- rep(names(sc_data)[i], ncol(sc_data[[i]]))
  gene_overlap <- intersect(rownames(sce), rownames(sc_data[[i]]))
  coldata_overlap <- intersect(names(colData(sce)), names(colData(sc_data[[i]])))
  sc_data[[i]] <- sc_data[[i]][gene_overlap,]
  colData(sc_data[[i]]) <- colData(sc_data[[i]])[, coldata_overlap]
  colData(sce) <- colData(sce)[, coldata_overlap]
  sce <- sce[gene_overlap,]
  sce <- cbind(sce, sc_data[[i]])
}

colnames(sce) <- paste0(colnames(sce), "_", sce$protocol)

sce <- runPCA(sce, ncomponents = 20)
sce <- runTSNE(sce)
sce <- runUMAP(sce)

#colors
cols <-c(c(jcolors('pal6'),jcolors('pal8'))[c(1,8,14,5,2:4,6,7,9:13,15:20)],jcolors('pal4'))
names(cols) <- c()

#param
MultiSample = FALSE

#variables
batch <- "protocol"
celltype <- "cell_line"
sample <- NA

table(colData(sce)[, batch])

#contrast
cont<-list("sc_10x-sc_celseq" = c(1,-1,0),
           "sc_10x-sc_dropseq" = c(1,0,-1),
           "sc_celseq-sc_dropseq" = c(0,1,-1))

```

# Visualize batch effect
```{r vis batch}
feature_list <- c(batch, "number_of_genes", celltype)

lapply(feature_list, function(feature_name){
  visGroup(sce, feature_name, dim_red= "UMAP")
})

```

# Size/Strength of the batch effcet
How much of the variance within data can be attributed to the batch effect?

## VariancePartitioning
```{r variance part}
expr <- as.matrix(assays(sce)$logcounts)
meta_sub <- as.data.frame(colData(sce)[, c(celltype, batch)])
form <- as.formula(paste0("~ (1|", celltype, ") + (1|", batch, ")"))
#varPart <- fitExtractVarPartModel(expr, form, meta_sub)

#Sort variables (i.e. columns) by median fraction# of variance explained
#vp <- varPart
#saveRDS(vp, paste0(out_path,"/vp_", dataset_name, ".rds"))
vp <- readRDS(paste0(out_path,"/vp_", dataset_name, ".rds"))

vp_names <- rownames(vp)
vp <-vp  %>% dplyr::mutate(gene= vp_names) %>% dplyr::arrange(-!! rlang::parse_expr(batch))
vp_sub <- vp[1:3] %>% set_rownames(vp$gene)

#plot
plotPercentBars( vp_sub[1:10,] )
plotVarPart( vp_sub )

```



Summarize variance partitioning

```{r summarize}
#How many genes have a variance component affected by the batch variable with more than 1%
n_batch_gene <- as_tibble(vp) %>%  dplyr::filter(!! rlang::parse_expr(batch) > 0.01) %>% nrow()
n_batch_gene10 <- as_tibble(vp) %>%  dplyr::filter(!! rlang::parse_expr(batch) > 0.1) %>% nrow()
n_celltype_gene <- as_tibble(vp) %>%  dplyr::filter(!! rlang::parse_expr(celltype)> 0.01) %>% nrow()
n_rel <- n_batch_gene/n_celltype_gene

#The mean percentage of the variance that is explained by the batch effect independent from the celltype
m_batch <- mean(vp[,batch])
m_celltype <- mean(vp[,celltype])
m_rel <- m_batch/m_celltype


#The median percentage of the variance that is explained by the batch effect independent from the celltype
me_batch <- median(vp[, batch])
me_celltype <- median(vp[, celltype])
me_rel <- me_batch/me_celltype
```


# Celltype specificity

## Celltype abundance
```{r celltype abundance}
meta_tib <- as_tibble(colData(sce)) %>% group_by_at(c(batch, celltype)) %>% summarize(n = n()) %>% dplyr::mutate(cell_freq = n / sum(n))


plot_abundance <- function(cluster_var, tib, x_var){
  meta_df <- as.data.frame(eval(tib))
  p <-ggplot(data=meta_df, aes_string(x=x_var, y="cell_freq", fill = cluster_var)) +
    geom_bar(stat="identity") + scale_fill_manual(values=cols)
  p + coord_flip() + theme_minimal()
}

plot_abundance(cluster_var = celltype, tib = meta_tib, x_var = batch)
```


## Celltype abundance - relative difference in abundances

```{r celltype relative diffrenece in abundance}
meta_tib <- as_tibble(colData(sce)) %>% group_by_at(c(batch, celltype)) %>% 
  summarize(n = n()) %>% spread_(batch,"n")
meta_df <- as.data.frame(eval(meta_tib))[,-1]
meta_comb<-combn(meta_df,2,simplify=FALSE)
res<-lapply(meta_comb,function(x){
  cond1<-names(x)[1]
  cond2<-names(x)[2]
  rel_abund_diff<-mapply(function(cond1,cond2) abs(cond1-cond2)/(cond1+cond2), x[,cond1], x[,cond2])
  rel_abund_diff
  })

mean_rel_abund_diff<-mean(unlist(res))
min_rel_abund_diff<-min(unlist(res))
max_rel_abund_diff<-max(unlist(res))
```

Test for significant changes in celltype abundances. Diffcyt (skip for datasets without replicates)
```{r sum}
if(MultiSample){
  #colData
  col_tib <- as_tibble(colData(sce)) %>% group_by_at(c(sample, batch)) %>% summarize()
  col_dat <- data.frame("sample_id" = col_tib[,sample], "group_id" = col_tib[,batch]) %>% set_rownames(levels(as.factor(colData(sce)[,sample])))
  
  
  #create desing matrix
  design <- createDesignMatrix(
    col_dat, cols_design = c(batch)
  )
  
  #create contrast
  n_contrast <- length(levels(as.factor(colData(sce)[,batch])))
  contrast <- diag(n_contrast)
  
  #create summarizedExperiment for cluster abundance (rows = cluster, columns= samples)
  cluster_counts <- as_tibble(colData(sce)) %>% group_by_at(c(sample, celltype)) %>% summarize(n=n()) %>%  spread_(sample, "n")
  cluster_count <- as.matrix(cluster_counts[,-1])
  rownames(cluster_count) <- unlist(cluster_counts[, celltype])
  
  #rowData
  row_dat <- as_tibble(colData(sce)) %>% group_by_at(celltype) %>% summarize(n=n()) %>% set_colnames(c("cluster_id", "n_cells"))
  
  
  
  se <- SummarizedExperiment(list("counts" = cluster_count), rowData= as.data.frame(row_dat), colData= col_dat)
  
  
  #differential abundance
  res_DA <- testDA_edgeR(se, design, contrast)
  
  #summarize - number of differential abundance cluster
  n_da_cluster <- length(which(rowData(res_DA)$p_adj < 0.01))
  per_da_cluster <- n_da_cluster/length(levels(as.factor(colData(sce)[,celltype])))
  
}else{
  per_da_cluster <- NA
  n_da_cluster<-0
}

```

# Count distribution
Do the overall counts distribution vary between batches? Clusterwise ploting?

```{r count distributions, fig.width = 7, fig.height = 6}
#sample an cluster ids
sids <- levels(as.factor(colData(sce)[, batch]))
names(sids) <- sids
cids <- levels(as.factor(colData(sce)[, celltype]))
names(cids) <- cids

#mean gene expression by sample and cluster
mean_list <- lapply(sids, function(batch_var){
  mean_cluster <- lapply(cids, function(cluster_var){
    counts_sc <- as.matrix(logcounts(
      sce[, colData(sce)[, batch] %in% batch_var & colData(sce)[, celltype] %in% cluster_var]))
  })
  mean_c <- mean_cluster %>% map(rowMeans) %>% bind_rows %>%
    dplyr::mutate(gene=rownames(sce)) %>% gather(cluster, logcounts, levels(as.factor(colData(sce)[,celltype])))
})

mean_expr <- mean_list %>% bind_rows(.id= "sample")

ggplot(mean_expr, aes(x=logcounts, colour=sample)) + geom_density(alpha=.3) +
  theme_classic() +
  facet_wrap( ~ cluster, ncol = 3) +
  scale_colour_manual(values = cols[c(1:3,7)]) +
  scale_x_continuous(limits =  c(0, 7))

#number of cluster with differnt batch distributions
per_dist <- 1
```


Mean expression protocols vs each other
```{r mean expression, fig.height=6}

#mean expression
mean_expr <- mean_list %>% bind_rows(.id = "sample" ) %>% spread(sample, logcounts)
batch_all <- levels(as.factor(colData(sce)[,batch]))

lapply(batch_all, function(batch_var){
  batch_var_2 <- batch_all[-which(batch_all %in% batch_var)]
  lapply(batch_var_2, function(batch_var_3){
    ggplot(mean_expr, aes_string(x=batch_var, y=batch_var_3)) +
      geom_point(alpha=.3, aes(color=cluster)) +
      ggtitle(batch_var) + geom_abline(slope = 1) + coord_fixed() +
      facet_wrap( ~ cluster, ncol = 3) +
      scale_color_manual(values=cols)
  })
})



```

# Cellspecific mixing score
Calculate cms
```{r cms}
# sce <- cms(sce, group = batch, k = 120, cell_min = 10, n_dim = 10)
#saveRDS(sce, paste0(out_path, "/cms_", dataset_name, ".rds"))
sce <- readRDS(paste0(out_path, "/cms_", dataset_name, ".rds"))

visHist(sce, n_col = 2)
visMetric(sce, metric_var = "cms_smooth", dim_red = "UMAP")

#summarize
mean_cms <- mean(sce$cms)
n_cms_0.01 <- length(which(sce$cms < 0.01))
cluster_mean_cms <- as_tibble(colData(sce)) %>% group_by_at(celltype) %>% summarize(cms_mean = mean(cms))

var_cms <- sd(cluster_mean_cms$cms_mean)

```
# DE analysis
Calculate DEG
```{r deg}
x<-sce
clust <- as.factor(colData(sce)[,celltype])
# n_cells <- table(clust,colData(sce)[,sample])
kids<-levels(clust)
names(kids) <- kids
group<-as.factor(colData(sce)[,batch])

cs <- names(cont)
names(cs) <- cs
es<- expr
ctype <- "contrast"
res_df <- function(k, tt, ct, c) {
  df <- data.frame(
    gene = rownames(tt), cluster_id = k, tt,
    row.names = NULL, stringsAsFactors = FALSE)
  df[[ct]] <- c
  return(df)
}
doDE <- function(x,lfc_cutoff=log2(1.1)){
  res<-lapply(kids, function (k) {
    cat(k, "..", sep = "")
    n <- clust==k
    es_tmp <- es[,n]
    grp <- group[n]
    design <- model.matrix(~0+grp)
    colnames(design)<-levels(group)
    k1 <- rowSums(es_tmp > 0) >= .2*min(table(grp))
    es_tmp <- es_tmp[k1,]
    f <- lmFit(es_tmp, design)
    f <- eBayes(f, trend = TRUE)
    tt <- lapply(cont, function(c) {
      cc<-names(c)
      fc <- contrasts.fit(f, contrasts = c)
      tr <- treat(fc, lfc=lfc_cutoff)
      tt <- topTreat(tr, n=Inf)
      res_df(k, tt, ctype,cc)
    })
    return(list(tt = tt, data = es_tmp))
  })
  # remove empty clusters
  skipped <- vapply(res, is.null, logical(1))
  if (any(skipped))
    message(paste("Cluster(s)", dQuote(kids[skipped]), "skipped due to an",
                  "insufficient number of cells in at least 2 samples per group."))
  res <- res[!skipped]
  kids <- kids[names(res)]

  # re-organize by contrast &
  # do global p-value adjustment
  tt <- lapply(res, "[[", "tt")
  tt <- lapply(cs, function(c) map(tt, c))

  # return results
  data <- lapply(res, "[[", "data")
  list(table = tt,
       data = data,
       design = design,
       coef = coef)
}
# res<-doDE(x,lfc_cutoff=log2(1.1))
# saveRDS(res, file=paste0(out_path, "/limma_", dataset_name, ".rds"))
```

## Some diagnostics plot of DEG

# Upset plot
```{r upset, fig.height=6}

res<-readRDS(paste0(out_path, "/limma_", dataset_name, ".rds"))
# count nb. of DE genes by cluster
# vapply(res[[1]][[2]], function(x) sum(x$adj.P.Val < 0.05), numeric(1))

FilterDEGs<-function (degDF=df, filter=c(FDR=5))
{
  rownames(degDF)<-degDF$gene
  pval <- degDF[, grep("adj.P.Val$", colnames(degDF)), drop = FALSE]
  pf <- pval <= filter["FDR"]/100
  pf[is.na(pf)] <- FALSE
  DEGlistUPorDOWN <- sapply(colnames(pf), function(x) rownames(pf[pf[, x, drop = FALSE], , drop = FALSE]), simplify = FALSE)
}

result<-list()
m2<-list()

for(jj in 1:length(cs)){
    result[[jj]]<-sapply(res[[1]][[names(cs)[jj]]], function(x) FilterDEGs(x))
    names(result[[jj]])<-kids
    m2[[jj]] = make_comb_mat(result[[jj]], mode = "intersect")
}
names(result)<-names(cs)
names(m2)<-names(cs)
# saveRDS(m2, file=paste0(out_path, "/upset_", dataset_name, ".rds"))
m2<-readRDS(paste0(out_path, "/upset_", dataset_name, ".rds"))


lapply(m2, function(x) UpSet(x))


```
# Histograms of p-values

```{r histograms, fig.height=6}
source(paste0(code_path, "/diagnostic_plots.R"))
histogram.gg(res)
baloonplot.gg(result)
```

# ratio specific cluster of DEG
```{r ratio_cluster_specific, fig.height=6}

n_de<-lapply(res[[1]],function(y) vapply(y, function(x) sum(x$adj.P.Val < 0.05), numeric(1)))
mean_n_de<-lapply(n_de,function(x) mean(x))
mean_mean_n_de<-mean(unlist(mean_n_de))
min_mean_n_de<-min(unlist(mean_n_de))
max_mean_n_de<-max(unlist(mean_n_de))
n_genes_lfc1<-lapply(res[[1]],function(y) vapply(y, function(x) sum(abs(x$logFC) > 1), numeric(1)))
mean_n_genes_lfc1<-mean(unlist(n_genes_lfc1))
min_n_genes_lfc1<-min(unlist(n_genes_lfc1))
max_n_genes_lfc1<-max(unlist(n_genes_lfc1))

unique_genes<-numeric()

de_overlap<-lapply(result,function(x){
  which.Null<-lapply(x,function(y) is.null(y))
  ind<-which.Null=='FALSE'
  result2<-x[ind]
  de_overlap<-length(Reduce(intersect, result2))
  de_overlap
})
mean_de_overlap<-mean(unlist(de_overlap))
min_de_overlap<-min(unlist(de_overlap))
max_de_overlap<-max(unlist(de_overlap))
unique_genes_matrix<-NULL
unique_genes<-NULL
cb<-length(names(result[[1]]))
unique_genes<-lapply(result,function(x){
  for(i in 1:cb){
    unique_genes[i]<-as.numeric(length(setdiff(unlist(x[i]),unlist(x[-i]))))
  }
  unique_genes_matrix<-cbind(unique_genes_matrix,unique_genes)
  unique_genes_matrix
})
unique_genes<-Reduce('cbind', unique_genes)
colnames(unique_genes)<-names(result)
rownames(unique_genes)<-names(result[[1]])

rel_spec1<-NULL
for(i in 1:dim(unique_genes)[2]){
    rel_spec<-unique_genes[,i]/de_overlap[[i]]
    rel_spec1<-cbind(rel_spec1,rel_spec)
}

mean_rel_spec=mean(rel_spec1)
min_rel_spec=min(rel_spec1)
max_rel_spec=max(rel_spec1)
```



# Summarize results
```{r summary}
#Size? How much of the variance can be attributed to the batch effect?
size <- data.frame("batch_genes_1per" = n_batch_gene,
                   "batch_genes_10per" = n_batch_gene10,
                   "celltype_gene_1per" = n_celltype_gene,
                   "relative_batch_celltype" = n_rel,
                   "mean_var_batch" = m_batch,
                   "mean_var_celltype" = m_celltype,
                   "median_var_batch" = me_batch,
                   "median_var_celltype" = me_celltype)


#Celltype-specificity? How celltype/cluster specific are batch effects? Differences in sample variation between batches?
celltype <- data.frame("DA_celltypes" = n_da_cluster,
                       "per_count_dist" = per_dist,
                       "mean_cms" = mean_cms,
                       'mean_rel_abund_diff' = mean_rel_abund_diff,
                       'min_rel_abund_diff' = min_rel_abund_diff,
                       'max_rel_abund_diff' = max_rel_abund_diff,
                       "celltype_var_cms" = var_cms,
                       "n_cells_cms_0.01" = n_cms_0.01)

#Gene-specificity? How do they effect genes? Single genes? All genes? Which genes?
gene <- data.frame("mean_mean_n_de_genes" = mean_mean_n_de,
                    "max_mean_n_de_genes" = max_mean_n_de,
                    "min_mean_n_de_genes" = min_mean_n_de,
                    "mean_n_genes_lfc1" = mean_n_genes_lfc1,
                    "min_n_genes_lfc1" = min_n_genes_lfc1,
                    "max_n_genes_lfc1" = max_n_genes_lfc1,
                     "mean_de_overlap" = mean_de_overlap,
                   "min_de_overlap" = min_de_overlap,
                   "max_de_overlap" = max_de_overlap,
                  "mean_rel_cluster_spec"= mean_rel_spec,
                  "min_rel_cluster_spec"= min_rel_spec,
                  "max_rel_cluster_spec"= max_rel_spec
                  )
# Cell-specificity? How cell-specific are batche effects? Are their differences in within celltype variation between batches?

summary <- cbind(size, celltype, gene) %>% set_rownames(dataset_name)
saveRDS(summary, paste0(out_path, "/summary_", dataset_name, ".rds"))

```


