---
title: "cellbench_seq_protocols"
author: "almutlue"
date: "2019-05-29"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Technical batch effects
**Cellbench data**, test batch effect from different protocols. Using the same three celllines sequenced by sequencing methods.
This represents a technical batch effect. 

libraries
```{r libs}
suppressPackageStartupMessages({
    library(CellBench)
    library(scater)
    library(jcolors)
    library(CellMixS)
    library(gridExtra)
    library(purrr)
    library(jcolors)
    library(here)
    library(tidyr)
    library(dplyr)
    library(stringr)
    library(variancePartition)
    library(diffcyt)
})
```

Dataset
```{r load data}
# data 
out_path <- here::here("output")
dataset_name <- "cellBench"

#Mixology dataset
sc_data <- load_sc_data()
colData(sc_data[[1]])$protocol <- rep(names(sc_data)[1], ncol(sc_data[[1]]))
sce <- sc_data[[1]]

for(i in 2:length(sc_data)){
  colData(sc_data[[i]])$protocol <- rep(names(sc_data)[i], ncol(sc_data[[i]]))
  gene_overlap <- intersect(rownames(sce), rownames(sc_data[[i]]))
  coldata_overlap <- intersect(names(colData(sce)), names(colData(sc_data[[i]]))) 
  sc_data[[i]] <- sc_data[[i]][gene_overlap,]
  colData(sc_data[[i]]) <- colData(sc_data[[i]])[, coldata_overlap]
  colData(sce) <- colData(sce)[, coldata_overlap]
  sce <- sce[gene_overlap,]
  sce <- cbind(sce, sc_data[[i]])
}

colnames(sce) <- paste0(colnames(sce), "_", sce$protocol)

sce <- runPCA(sce, ncomponents = 20)
sce <- runTSNE(sce)
sce <- runUMAP(sce)

#colors
cols <-c(c(jcolors('pal6'),jcolors('pal8'))[c(1,8,14,5,2:4,6,7,9:13,15:20)],jcolors('pal4'))
names(cols) <- c()

#param
MultiSample = FALSE

#variables
batch <- "protocol"
celltype <- "cell_line"
sample <- NA

```

# Visualize batch effect
```{r vis batch}
feature_list <- c(batch, "number_of_genes", celltype)

lapply(feature_list, function(feature_name){
  visGroup(sce, feature_name, dim_red= "UMAP")
})

```

# Size/Strength of the batch effcet
How much of the variance within data can be attributed to the batch effect?

## VariancePartitioning
```{r variance part}
expr <- as.matrix(assays(sce)$logcounts)
meta_sub <- as.data.frame(colData(sce)[, c(celltype, batch)])
form <- as.formula(paste0("~ (1|", celltype, ") + (1|", batch, ")"))
#varPart <- fitExtractVarPartModel(expr, form, meta_sub) 

#Sort variables (i.e. columns) by median fraction# of variance explained
#vp <- sortCols(varPart)
#saveRDS(vp, paste0(out_path,"/vp_", dataset_name, ".rds"))
vp <- readRDS(paste0(out_path,"/vp_", dataset_name, ".rds"))

vp_names <- rownames(vp)
vp <-vp  %>% dplyr::mutate(gene= vp_names) %>% dplyr::arrange(-!! rlang::parse_expr(batch))
vp_sub <- vp[1:3] %>% set_rownames(vp$gene)

#plot 
plotPercentBars( vp_sub[1:10,] )
plotVarPart( vp_sub )

```



Summarize variance partitioning
```{r summarize}
#How many genes have a variance component affected by the batch variable with more than 1%
n_batch_gene <- as_tibble(vp) %>%  dplyr::filter(!! rlang::parse_expr(batch) > 0.01) %>% nrow()
n_batch_gene10 <- as_tibble(vp) %>%  dplyr::filter(!! rlang::parse_expr(batch) > 0.1) %>% nrow()
n_celltype_gene <- as_tibble(vp) %>%  dplyr::filter(!! rlang::parse_expr(celltype)> 0.01) %>% nrow()
n_rel <- n_batch_gene/n_celltype_gene

#The mean percentage of the variance that is explained by the batch effect independent from the celltype
m_batch <- mean(vp[,batch])
m_celltype <- mean(vp[,celltype])
m_rel <- m_batch/m_celltype


#The median percentage of the variance that is explained by the batch effect independent from the celltype
me_batch <- median(vp[, batch])
me_celltype <- median(vp[, celltype])
me_rel <- me_batch/me_celltype
```


# Celltype specificity

## Celltype abundance
```{r celltype abundance}
meta_tib <- as_tibble(colData(sce)) %>% group_by_at(c(batch, celltype)) %>% summarize(n = n()) %>% dplyr::mutate(cell_freq = n / sum(n))


plot_abundance <- function(cluster_var, tib, x_var){
  meta_df <- as.data.frame(eval(tib))
  p <-ggplot(data=meta_df, aes_string(x=x_var, y="cell_freq", fill = cluster_var)) +
  geom_bar(stat="identity") + scale_fill_manual(values=cols)
  p + coord_flip() + theme_minimal()
}

plot_abundance(cluster_var = celltype, tib = meta_tib, x_var = batch)
```

Test for significant changes in celltype abundances. Diffcyt (skip for datasets without replicates)
```{r sum}
if(MultiSample){
  #colData

col_dat <-  data.frame("sample_id" = levels(colData(sce)[,sample]), "group_id" = levels(colData(sce)[,batch])) %>% set_rownames(levels(colData(sce)[,sample])) 


#create desing matrix
design_ <- createDesignMatrix(
  col_dat, cols_design = c("group_id")
)

#create contrast
contrast <- createContrast(c(0, 1))

#create summarizedExperiment for cluster abundance (rows = cluster, columns= samples)
cluster_counts <- as_tibble(colData(sce)) %>% group_by_at(c(sample, batch)) %>% summarize(n=n()) %>%  spread(sample_id, n) 
cluster_count <- as.matrix(cluster_counts[,-1])
rownames(cluster_count) <- cluster_counts[, celltype]

#rowData
row_dat <- as_tibble(colData(sce)) %>% group_by_at(celltype) %>% summarize(n=n()) %>% set_colnames(c("cluster_id", "n_cells"))



se <- SummarizedExperiment(list("counts" = cluster_count), rowData= as.data.frame(row_dat), colData= col_dat)


#differential abundance
res_DA <- testDA_edgeR(se, design_, contrast)

#summarize - number of differential abundance cluster
n_da_cluster <- length(which(rowData(res_DA)$p_adj < 0.1))

}else{
  n_da_cluster <- NA
}

```

# Count distribution
Do the overall counts distribution vary between batches? Clusterwise ploting?

```{r count distributions, fig.width = 7, fig.height = 6}
#sample an cluster ids
sids <- levels(as.factor(colData(sce)[, batch]))
names(sids) <- sids
cids <- levels(as.factor(colData(sce)[, celltype]))
names(cids) <- cids

#mean gene expression by sample and cluster
mean_list <- lapply(sids, function(batch_var){
  mean_cluster <- lapply(cids, function(cluster_var){
    counts_sc <- as.matrix(logcounts(
        sce[, colData(sce)[, batch] %in% batch_var & colData(sce)[, celltype] %in% cluster_var]))
  })
  mean_c <- mean_cluster %>% map(rowMeans) %>% bind_rows %>%
    dplyr::mutate(gene=rownames(sce)) %>% gather(cluster, logcounts, levels(as.factor(colData(sce)[,celltype]))) 
  })

mean_expr <- mean_list %>% bind_rows(.id= "sample") 

ggplot(mean_expr, aes(x=logcounts, colour=sample)) + geom_density(alpha=.3) + 
    theme_classic() +
    facet_wrap( ~ cluster, ncol = 3) +
    scale_colour_manual(values = cols[c(1:3,7)]) +
    scale_x_continuous(limits =  c(0, 7))

#number of cluster with differnt batch distributions
per_dist <- 1
```


Mean expression protocols vs each other
```{r mean expression, fig.height=6}
  
#mean expression
mean_expr <- mean_list %>% bind_rows(.id = "sample" ) %>% spread(sample, logcounts)
batch_all <- levels(as.factor(colData(sce)[,batch]))
  
lapply(batch_all, function(batch_var){
  batch_var_2 <- batch_all[-which(batch_all %in% batch_var)]
  lapply(batch_var_2, function(batch_var_3){
      ggplot(mean_expr, aes_string(x=batch_var, y=batch_var_3)) + 
      geom_point(alpha=.3, aes(color=cluster)) + 
      ggtitle(batch_var) + geom_abline(slope = 1) + coord_fixed() +
      facet_wrap( ~ cluster, ncol = 3) +
      scale_color_manual(values=cols)
  })
})
  

  
```

# Cellspecific mixing score
Calculate cms
```{r cms}
sce <- cms(sce, group = batch, k = 120, cell_min = 10, n_dim = 10)
#saveRDS(sce, paste0(out_path, "/cms_", dataset_name, ".rds"))
#sce <- readRDS(paste0(out_path, "/pbmc_media_cms.rds"))

visHist(sce, n_col = 2)
visMetric(sce, metric_var = "cms_smooth", dim_red = "UMAP")

#summarize
mean_cms <- mean(sce$cms)
n_cms_0.01 <- length(which(sce$cms < 0.01))
cluster_mean_cms <- as_tibble(colData(sce)) %>% group_by_at(celltype) %>% summarize(cms_mean = mean(cms))

var_cms <- sd(cluster_mean_cms$cms_mean)

```


# Summarize results
```{r summary}
#Size? How much of the variance can be attributed to the batch effect?
size <- data.frame("batch_genes_1per" = n_batch_gene,  
                   "batch_genes_10per" = n_batch_gene10,
                   "celltype_gene_1per" = n_celltype_gene,
                   "relative_batch_celltype" = n_rel,
                   "mean_var_batch" = m_batch,
                   "mean_var_celltype" = m_celltype,
                   "median_var_batch" = me_batch,
                   "median_var_celltype" = me_celltype)


#Celltype-specificity? How celltype/cluster specific are batch effects? Differences in sample variation between batches?
celltype <- data.frame("DA_celltypes" = n_da_cluster,
                       "per_count_dist" = per_dist,
                       "mean_cms" = mean_cms,
                       "celltype_var_cms" = var_cms,
                       "n_cells_cms_0.01" = n_cms_0.01)

#Gene-specificity? How do they effect genes? Single genes? All genes? Which genes?
#gene <- data.frame("mean_n_de_genes" = mean_n_de,
#                   "n_genes_lfc2" = n_genes_lfc2,
#                   "DE_overlap" = de_overlap,
#                   "DE_overlap50" = de_overlap50,
#                   "rel_cluster_specific_DE" = rel_spec
#                   )
# Cell-specificity? How cell-specific are batche effects? Are their differences in within celltype variation between batches?

summary <- cbind(size, celltype )#, gene) %>% set_rownames("pbmc_media_storage")
saveRDS(summary, paste0(out_path, "/summary_pbmc_media_storage.rds"))

```

